{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "328b5c82",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "f25bae01",
   "metadata": {},
   "source": [
    "# GWAS AD data (Rosmap)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "831d6b85",
   "metadata": {},
   "source": [
    "## 1. Obtaining dictionary of AD GWAS associated genes and the variant names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "490e6bd3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dictionary saved to gene_variants_dict_AD.pkl\n"
     ]
    }
   ],
   "source": [
    "# I don't have these files anymore on the server\n",
    "# But they are available on the mac\n",
    "\n",
    "import os\n",
    "import pandas as pd\n",
    "import pickle\n",
    "\n",
    "# Directory containing the files\n",
    "directory = './gwas_genes_dict/'\n",
    "\n",
    "# Initialize an empty dictionary to store the gene to variants mapping\n",
    "gene_variants_dict = {}\n",
    "\n",
    "# Iterate through all files in the directory\n",
    "for filename in os.listdir(directory):\n",
    "    if filename.endswith('.txt'):\n",
    "        filepath = os.path.join(directory, filename)\n",
    "        # Read the file into a DataFrame\n",
    "        df = pd.read_csv(filepath, sep='\\t')\n",
    "        \n",
    "        # Iterate through each row in the DataFrame\n",
    "        for index, row in df.iterrows():\n",
    "            gene_id = row['Gene stable ID']\n",
    "            variant_name = row['Variant name']\n",
    "            \n",
    "            # Add the variant to the list of variants for the gene in the dictionary\n",
    "            if gene_id not in gene_variants_dict:\n",
    "                gene_variants_dict[gene_id] = []\n",
    "            gene_variants_dict[gene_id].append(variant_name)\n",
    "\n",
    "# Save the dictionary to a pickle file\n",
    "output_pickle_file = 'gene_variants_dict_AD.pkl'\n",
    "# with open(output_pickle_file, 'wb') as f:\n",
    "#     pickle.dump(gene_variants_dict, f)\n",
    "\n",
    "print(f\"Dictionary saved to {output_pickle_file}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a9d739d6",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_variants_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "efbeaaea",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'gene_variants_dict' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[6], line 3\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;21;01mpickle\u001b[39;00m\n\u001b[1;32m      2\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mopen\u001b[39m(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mad_shap_gwas/snps_list/gene_variants_dict_AD_filt_phenotype.pkl\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mwb\u001b[39m\u001b[38;5;124m'\u001b[39m) \u001b[38;5;28;01mas\u001b[39;00m f:\n\u001b[0;32m----> 3\u001b[0m     pickle\u001b[38;5;241m.\u001b[39mdump(\u001b[43mgene_variants_dict\u001b[49m, f)\n",
      "\u001b[0;31mNameError\u001b[0m: name 'gene_variants_dict' is not defined"
     ]
    }
   ],
   "source": [
    "import pickle\n",
    "with open('ad_shap_gwas/snps_list/gene_variants_dict_AD_filt_phenotype.pkl', 'wb') as f:\n",
    "    pickle.dump(gene_variants_dict, f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "f051c41a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of keys (gene IDs) in the dictionary: 113\n",
      "Average number of variants per gene: 46106.13\n"
     ]
    }
   ],
   "source": [
    "# Calculate the number of keys in the dictionary\n",
    "num_keys = len(gene_variants_dict)\n",
    "print(f\"Number of keys (gene IDs) in the dictionary: {num_keys}\")\n",
    "\n",
    "# Calculate the mean number of values (variants) per key in the dictionary\n",
    "if num_keys > 0:\n",
    "    mean_values = sum(len(variants) for variants in gene_variants_dict.values()) / num_keys\n",
    "    print(f\"Average number of variants per gene: {mean_values:.2f}\")\n",
    "else:\n",
    "    print(\"The dictionary is empty.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f2b01fa4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "4506c051",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pickle\n",
    "with open('ad_shap_gwas/snps_list/gene_variants_dict_AD.pkl', 'rb') as f:\n",
    "    gene_variants_dict = pickle.load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d123ed69",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_variants_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "57e1188c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cd61c7f6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filtering the dict to reduce number of snps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "085bd5ca",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total SNPs in dictionary: 5209993\n",
      "Matching SNPs: 24346\n",
      "Percentage of Matching SNPs: 0.47%\n"
     ]
    }
   ],
   "source": [
    "# How many snps are the same as phenotype associated\n",
    "\n",
    "import pickle\n",
    "\n",
    "# Load the SNP list from the text file\n",
    "snp_list_path = './all_snps_list_phenotype_associated.txt'\n",
    "with open(snp_list_path, 'r') as file:\n",
    "    snp_set = set(file.read().splitlines())\n",
    "\n",
    "# Count how many SNPs in the dictionary are in the SNP list\n",
    "matching_snps_count = 0\n",
    "total_snps_count = 0\n",
    "\n",
    "for variants in gene_variants_dict.values():\n",
    "    for snp in variants:\n",
    "        total_snps_count += 1\n",
    "        if snp in snp_set:\n",
    "            matching_snps_count += 1\n",
    "\n",
    "print(f\"Total SNPs in dictionary: {total_snps_count}\")\n",
    "print(f\"Matching SNPs: {matching_snps_count}\")\n",
    "print(f\"Percentage of Matching SNPs: {(matching_snps_count / total_snps_count * 100):.2f}%\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "c834e1a5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total SNPs in dictionary: 5209993\n",
      "Matching SNPs: 1928\n",
      "Percentage of Matching SNPs: 0.04%\n"
     ]
    }
   ],
   "source": [
    "# How many snps are the same as phenotype associated\n",
    "\n",
    "import pickle\n",
    "\n",
    "# Load the SNP list from the text file\n",
    "snp_list_path = './ad_shap_gwas/snps_list/all_snps_list_clinically_associated.txt'\n",
    "with open(snp_list_path, 'r') as file:\n",
    "    snp_set = set(file.read().splitlines())\n",
    "\n",
    "# Count how many SNPs in the dictionary are in the SNP list\n",
    "matching_snps_count = 0\n",
    "total_snps_count = 0\n",
    "\n",
    "for variants in gene_variants_dict.values():\n",
    "    for snp in variants:\n",
    "        total_snps_count += 1\n",
    "        if snp in snp_set:\n",
    "            matching_snps_count += 1\n",
    "\n",
    "print(f\"Total SNPs in dictionary: {total_snps_count}\")\n",
    "print(f\"Matching SNPs: {matching_snps_count}\")\n",
    "print(f\"Percentage of Matching SNPs: {(matching_snps_count / total_snps_count * 100):.2f}%\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5e505092",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "aa3ff18d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filtered dictionary saved to gene_variants_dict_AD_filt_clinical.pkl\n"
     ]
    }
   ],
   "source": [
    "import pickle\n",
    "\n",
    "# Load the SNP list from the text file\n",
    "snp_list_path = './ad_shap_gwas/snps_list/all_snps_list_clinically_associated.txt'\n",
    "with open(snp_list_path, 'r') as file:\n",
    "    snp_set = set(file.read().splitlines())\n",
    "\n",
    "# Filter the dictionary to only include SNPs that are in the snp_set\n",
    "filtered_dict = {}\n",
    "for gene, variants in gene_variants_dict.items():\n",
    "    filtered_snps = [snp for snp in variants if snp in snp_set]\n",
    "    if filtered_snps:  # Only add gene if there are any matching SNPs\n",
    "        filtered_dict[gene] = filtered_snps\n",
    "\n",
    "# Save the filtered dictionary to a new pickle file\n",
    "filtered_pickle_file = 'gene_variants_dict_AD_filt_clinical.pkl'\n",
    "with open(filtered_pickle_file, 'wb') as f:\n",
    "    pickle.dump(filtered_dict, f)\n",
    "\n",
    "print(f\"Filtered dictionary saved to {filtered_pickle_file}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "191a46b8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "4913106b",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pickle\n",
    "with open('./ad_shap_gwas/snps_list/gene_variants_dict_AD_filt_phenotype.pkl', 'rb') as f:\n",
    "    filtered_dict_phen = pickle.load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "7c40638d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pickle\n",
    "with open('./ad_shap_gwas/snps_list/gene_variants_dict_AD_filt_clinical.pkl', 'rb') as f:\n",
    "    filtered_dict = pickle.load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "203e8ce0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of keys (gene IDs) in the dictionary: 55\n",
      "Average number of variants per gene: 35.05\n"
     ]
    }
   ],
   "source": [
    "# Calculate the number of keys in the dictionary\n",
    "num_keys = len(filtered_dict)\n",
    "print(f\"Number of keys (gene IDs) in the dictionary: {num_keys}\")\n",
    "\n",
    "# Calculate the mean number of values (variants) per key in the dictionary\n",
    "if num_keys > 0:\n",
    "    mean_values = sum(len(variants) for variants in filtered_dict.values()) / num_keys\n",
    "    print(f\"Average number of variants per gene: {mean_values:.2f}\")\n",
    "else:\n",
    "    print(\"The dictionary is empty.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "dbffd9ac",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of overlapping values: 1876\n"
     ]
    }
   ],
   "source": [
    "# Flatten the lists of values in each dictionary and convert to sets\n",
    "set1 = set(v for values in filtered_dict_phen.values() for v in values)\n",
    "set2 = set(v for values in filtered_dict.values() for v in values)\n",
    "\n",
    "# Calculate the intersection of the two sets\n",
    "overlap = set1.intersection(set2)\n",
    "\n",
    "# Print the results\n",
    "print(f\"Number of overlapping values: {len(overlap)}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "84864129",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ENSG00000073969': ['rs2146247619', 'rs2146266663'],\n",
       " 'ENSG00000103510': ['rs748699921',\n",
       "  'rs2057483219',\n",
       "  'rs2057538050',\n",
       "  'rs2057538160',\n",
       "  'rs2143966219',\n",
       "  'rs2143966454',\n",
       "  'rs2143985614'],\n",
       " 'ENSG00000137845': ['rs483352912',\n",
       "  'rs483352913',\n",
       "  'rs483352914',\n",
       "  'rs483352915',\n",
       "  'rs483352916'],\n",
       " 'ENSG00000198087': ['rs142643033',\n",
       "  'rs267606710',\n",
       "  'rs368795308',\n",
       "  'rs545551160',\n",
       "  'rs776297606',\n",
       "  'rs1393955970',\n",
       "  'rs1554181304',\n",
       "  'rs1769050790'],\n",
       " 'ENSG00000064687': ['rs189260652',\n",
       "  'rs115536223',\n",
       "  'rs538591288',\n",
       "  'rs551118463',\n",
       "  'rs547447016',\n",
       "  'rs149949633',\n",
       "  'rs200538373',\n",
       "  'rs770510230',\n",
       "  'rs779501556',\n",
       "  'rs761115996',\n",
       "  'rs777874744',\n",
       "  'rs2144710743',\n",
       "  'rs2144883649'],\n",
       " 'ENSG00000081014': ['rs542940704',\n",
       "  'rs141278078',\n",
       "  'rs556450190',\n",
       "  'rs569567659',\n",
       "  'rs757657323',\n",
       "  'rs760021635',\n",
       "  'rs754944429',\n",
       "  'rs773929706',\n",
       "  'rs868844377',\n",
       "  'rs1313275799',\n",
       "  'rs1567269045',\n",
       "  'rs1567230528',\n",
       "  'rs1555456727',\n",
       "  'rs1555462184',\n",
       "  'rs2064899680',\n",
       "  'rs2063801225',\n",
       "  'rs2141164712',\n",
       "  'rs2141164878',\n",
       "  'rs2141147450',\n",
       "  'rs2140861869',\n",
       "  'rs2140862092',\n",
       "  'rs2140878481',\n",
       "  'rs2140934862',\n",
       "  'rs2140934998'],\n",
       " 'ENSG00000089280': ['rs544088874',\n",
       "  'rs186547381',\n",
       "  'rs121909667',\n",
       "  'rs121909668',\n",
       "  'rs121909669',\n",
       "  'rs121909671',\n",
       "  'rs387906627',\n",
       "  'rs387906628',\n",
       "  'rs387907274',\n",
       "  'rs72550890',\n",
       "  'rs267606831',\n",
       "  'rs267606832',\n",
       "  'rs267606833',\n",
       "  'rs752076094',\n",
       "  'rs764487847',\n",
       "  'rs886041389',\n",
       "  'rs886041390',\n",
       "  'rs886041577',\n",
       "  'rs886041776',\n",
       "  'rs747579808',\n",
       "  'rs1161032867',\n",
       "  'rs1085308015',\n",
       "  'rs1228194239',\n",
       "  'rs1555509569',\n",
       "  'rs1555509609',\n",
       "  'rs1555509693',\n",
       "  'rs1596908744',\n",
       "  'rs1596912983',\n",
       "  'rs1596914113',\n",
       "  'rs1567479067',\n",
       "  'rs2079347087',\n",
       "  'rs2079350332',\n",
       "  'rs2079351113',\n",
       "  'rs2144138894',\n",
       "  'rs2144140268',\n",
       "  'rs2144142389'],\n",
       " 'ENSG00000095970': ['rs2234252',\n",
       "  'rs104893998',\n",
       "  'rs104894001',\n",
       "  'rs104894002',\n",
       "  'rs121908402',\n",
       "  'rs143332484',\n",
       "  'rs144250872',\n",
       "  'rs199910080',\n",
       "  'rs142232675',\n",
       "  'rs200392967',\n",
       "  'rs201258663',\n",
       "  'rs530314472',\n",
       "  'rs386834140',\n",
       "  'rs386834141',\n",
       "  'rs386834142',\n",
       "  'rs386834143',\n",
       "  'rs386834144',\n",
       "  'rs768621570',\n",
       "  'rs1294860969',\n",
       "  'rs1765488318',\n",
       "  'rs2113877345',\n",
       "  'rs2234252',\n",
       "  'rs104893998',\n",
       "  'rs104894001',\n",
       "  'rs104894002',\n",
       "  'rs121908402',\n",
       "  'rs143332484',\n",
       "  'rs144250872',\n",
       "  'rs199910080',\n",
       "  'rs142232675',\n",
       "  'rs200392967',\n",
       "  'rs201258663',\n",
       "  'rs530314472',\n",
       "  'rs386834140',\n",
       "  'rs386834141',\n",
       "  'rs386834142',\n",
       "  'rs386834143',\n",
       "  'rs386834144',\n",
       "  'rs768621570',\n",
       "  'rs1294860969',\n",
       "  'rs1765488318',\n",
       "  'rs2113877345'],\n",
       " 'ENSG00000103671': ['rs200549601',\n",
       "  'rs553667435',\n",
       "  'rs113596176',\n",
       "  'rs147303485',\n",
       "  'rs869312827',\n",
       "  'rs760447338',\n",
       "  'rs748828135',\n",
       "  'rs761865592',\n",
       "  'rs769545692',\n",
       "  'rs1446427502',\n",
       "  'rs1566972371',\n",
       "  'rs1596341256',\n",
       "  'rs1891715888',\n",
       "  'rs1900263561',\n",
       "  'rs2140297426',\n",
       "  'rs2140301381'],\n",
       " 'ENSG00000114805': ['rs757653002', 'rs2108014903'],\n",
       " 'ENSG00000142599': ['rs149900041',\n",
       "  'rs745806637',\n",
       "  'rs749071713',\n",
       "  'rs759541634',\n",
       "  'rs766951273',\n",
       "  'rs751232315',\n",
       "  'rs869312871',\n",
       "  'rs878853011',\n",
       "  'rs878853252',\n",
       "  'rs878853270',\n",
       "  'rs779300628',\n",
       "  'rs897627996',\n",
       "  'rs1057520747',\n",
       "  'rs1057524139',\n",
       "  'rs1064793252',\n",
       "  'rs1064795726',\n",
       "  'rs1233062866',\n",
       "  'rs1306272176',\n",
       "  'rs1321809020',\n",
       "  'rs1557461427',\n",
       "  'rs1557582259',\n",
       "  'rs1557582271',\n",
       "  'rs1557586047',\n",
       "  'rs1553153669',\n",
       "  'rs1553154130',\n",
       "  'rs1553154132',\n",
       "  'rs1553155581',\n",
       "  'rs1553155778',\n",
       "  'rs1641418583',\n",
       "  'rs1641481122',\n",
       "  'rs1641565884',\n",
       "  'rs1646008860',\n",
       "  'rs2124354095',\n",
       "  'rs2124359087',\n",
       "  'rs2124375440'],\n",
       " 'ENSG00000156170': ['rs137853184',\n",
       "  'rs201088736',\n",
       "  'rs201732170',\n",
       "  'rs552141388',\n",
       "  'rs575462405',\n",
       "  'rs749738738',\n",
       "  'rs756507716',\n",
       "  'rs758181982',\n",
       "  'rs762093523',\n",
       "  'rs762620949',\n",
       "  'rs753873681',\n",
       "  'rs768273248',\n",
       "  'rs765915512',\n",
       "  'rs868345341',\n",
       "  'rs863223931',\n",
       "  'rs863223932',\n",
       "  'rs1179490149',\n",
       "  'rs1057519084',\n",
       "  'rs1057519085',\n",
       "  'rs1057519086',\n",
       "  'rs1085307635',\n",
       "  'rs1224421127',\n",
       "  'rs1284681229',\n",
       "  'rs1463009913',\n",
       "  'rs1554669910',\n",
       "  'rs1830643251',\n",
       "  'rs1829405389'],\n",
       " 'ENSG00000159640': ['rs121912703',\n",
       "  'rs121912704',\n",
       "  'rs143507892',\n",
       "  'rs148491967',\n",
       "  'rs138873311',\n",
       "  'rs387906576',\n",
       "  'rs397514688',\n",
       "  'rs397514689',\n",
       "  'rs367797185',\n",
       "  'rs758933315',\n",
       "  'rs797045079',\n",
       "  'rs775421383',\n",
       "  'rs761345398',\n",
       "  'rs778390161',\n",
       "  'rs779188587',\n",
       "  'rs779422412',\n",
       "  'rs754265941',\n",
       "  'rs900084108',\n",
       "  'rs771053807',\n",
       "  'rs1005792910',\n",
       "  'rs1198993903',\n",
       "  'rs1219522144',\n",
       "  'rs1357918249',\n",
       "  'rs1341633213',\n",
       "  'rs1387186484',\n",
       "  'rs1409924291',\n",
       "  'rs1568038811',\n",
       "  'rs1599142041',\n",
       "  'rs1599143221',\n",
       "  'rs2030333357',\n",
       "  'rs2049629009',\n",
       "  'rs2049738010',\n",
       "  'rs2147530850',\n",
       "  'rs2147533687'],\n",
       " 'ENSG00000172500': ['rs786204849', 'rs886037928'],\n",
       " 'ENSG00000174469': ['rs187552025',\n",
       "  'rs73464271',\n",
       "  'rs78223661',\n",
       "  'rs117876038',\n",
       "  'rs121908445',\n",
       "  'rs138481453',\n",
       "  'rs188140291',\n",
       "  'rs143226012',\n",
       "  'rs143507886',\n",
       "  'rs143856702',\n",
       "  'rs144003099',\n",
       "  'rs144496909',\n",
       "  'rs139180845',\n",
       "  'rs150607716',\n",
       "  'rs147815978',\n",
       "  'rs148453565',\n",
       "  'rs149032771',\n",
       "  'rs149185385',\n",
       "  'rs146225600',\n",
       "  'rs146873712',\n",
       "  'rs141078449',\n",
       "  'rs141831869',\n",
       "  'rs142331907',\n",
       "  'rs142434274',\n",
       "  'rs369056998',\n",
       "  'rs369675346',\n",
       "  'rs370916766',\n",
       "  'rs371512835',\n",
       "  'rs371642222',\n",
       "  'rs371839994',\n",
       "  'rs199994339',\n",
       "  'rs200866893',\n",
       "  'rs201076428',\n",
       "  'rs201200400',\n",
       "  'rs201311931',\n",
       "  'rs201326295',\n",
       "  'rs201446615',\n",
       "  'rs201602527',\n",
       "  'rs267601384',\n",
       "  'rs368108883',\n",
       "  'rs56356283',\n",
       "  'rs377455159',\n",
       "  'rs398124268',\n",
       "  'rs535454043',\n",
       "  'rs540745201',\n",
       "  'rs372830287',\n",
       "  'rs562833882',\n",
       "  'rs546437079',\n",
       "  'rs727503878',\n",
       "  'rs730880275',\n",
       "  'rs730880276',\n",
       "  'rs745770134',\n",
       "  'rs746615217',\n",
       "  'rs746894028',\n",
       "  'rs751491210',\n",
       "  'rs752019151',\n",
       "  'rs752550849',\n",
       "  'rs749575477',\n",
       "  'rs749646225',\n",
       "  'rs753396304',\n",
       "  'rs760930032',\n",
       "  'rs761663690',\n",
       "  'rs757043221',\n",
       "  'rs758102050',\n",
       "  'rs758630057',\n",
       "  'rs772883080',\n",
       "  'rs773595457',\n",
       "  'rs770489662',\n",
       "  'rs770951811',\n",
       "  'rs771533907',\n",
       "  'rs771772546',\n",
       "  'rs771827120',\n",
       "  'rs794727872',\n",
       "  'rs796052374',\n",
       "  'rs796052388',\n",
       "  'rs781574808',\n",
       "  'rs886041991',\n",
       "  'rs886062049',\n",
       "  'rs776175685',\n",
       "  'rs776956365',\n",
       "  'rs778124698',\n",
       "  'rs766777011',\n",
       "  'rs767033703',\n",
       "  'rs767408882',\n",
       "  'rs767821521',\n",
       "  'rs972116002',\n",
       "  'rs986848347',\n",
       "  'rs1159246607',\n",
       "  'rs1057520468',\n",
       "  'rs1057520549',\n",
       "  'rs1057520743',\n",
       "  'rs1064796071',\n",
       "  'rs1064796221',\n",
       "  'rs1085307838',\n",
       "  'rs1176180722',\n",
       "  'rs1247068015',\n",
       "  'rs1396313317',\n",
       "  'rs1391540245',\n",
       "  'rs1433216677',\n",
       "  'rs1458942108',\n",
       "  'rs1554400338',\n",
       "  'rs1554416016',\n",
       "  'rs1554440668',\n",
       "  'rs1554490549',\n",
       "  'rs1584757170',\n",
       "  'rs1584848275',\n",
       "  'rs1585020100',\n",
       "  'rs1585151371',\n",
       "  'rs1794916576',\n",
       "  'rs1797490749',\n",
       "  'rs1800807147',\n",
       "  'rs1799910521',\n",
       "  'rs1799307071',\n",
       "  'rs1801395870',\n",
       "  'rs1801396588',\n",
       "  'rs1805809633',\n",
       "  'rs2116752243',\n",
       "  'rs2116789389',\n",
       "  'rs2116608214',\n",
       "  'rs2116638179',\n",
       "  'rs2116691068',\n",
       "  'rs2129185949',\n",
       "  'rs2129279519',\n",
       "  'rs2129279547',\n",
       "  'rs2129279551',\n",
       "  'rs2129279561',\n",
       "  'rs2116924115'],\n",
       " 'ENSG00000066336': ['rs773877800',\n",
       "  'rs1565638431',\n",
       "  'rs2095906404',\n",
       "  'rs2095906547',\n",
       "  'rs2095916574',\n",
       "  'rs2142884393'],\n",
       " 'ENSG00000069667': ['rs1057518981',\n",
       "  'rs1222860434',\n",
       "  'rs1198203108',\n",
       "  'rs1433850094',\n",
       "  'rs1555421544',\n",
       "  'rs1555423812',\n",
       "  'rs1555427497',\n",
       "  'rs1555427498',\n",
       "  'rs1595874842',\n",
       "  'rs1595874995',\n",
       "  'rs1595889010',\n",
       "  'rs2065374158',\n",
       "  'rs2065471243',\n",
       "  'rs2141286330',\n",
       "  'rs2141286878',\n",
       "  'rs2141319872',\n",
       "  'rs2141337837'],\n",
       " 'ENSG00000138600': ['rs1328617979', 'rs2141024987'],\n",
       " 'ENSG00000142192': ['rs145564988',\n",
       "  'rs63749810',\n",
       "  'rs63749953',\n",
       "  'rs63749964',\n",
       "  'rs63750064',\n",
       "  'rs63750066',\n",
       "  'rs63750264',\n",
       "  'rs63750363',\n",
       "  'rs63750399',\n",
       "  'rs63750579',\n",
       "  'rs63750643',\n",
       "  'rs63750671',\n",
       "  'rs63750734',\n",
       "  'rs63750921',\n",
       "  'rs63750973',\n",
       "  'rs63751039',\n",
       "  'rs137865262',\n",
       "  'rs193922916',\n",
       "  'rs199587668',\n",
       "  'rs199887707',\n",
       "  'rs199890425',\n",
       "  'rs201022619',\n",
       "  'rs201269325',\n",
       "  'rs201290605',\n",
       "  'rs281865161',\n",
       "  'rs2146237857',\n",
       "  'rs2146237965'],\n",
       " 'ENSG00000145700': ['rs1463928941', 'rs2150188446'],\n",
       " 'ENSG00000064666': ['rs77830704', 'rs78386506'],\n",
       " 'ENSG00000080815': ['rs661',\n",
       "  'rs199723282',\n",
       "  'rs121917807',\n",
       "  'rs121917808',\n",
       "  'rs121917809',\n",
       "  'rs63749805',\n",
       "  'rs63749806',\n",
       "  'rs63749824',\n",
       "  'rs63749836',\n",
       "  'rs63749880',\n",
       "  'rs63749885',\n",
       "  'rs63749891',\n",
       "  'rs63749925',\n",
       "  'rs63750001',\n",
       "  'rs63750004',\n",
       "  'rs63750009',\n",
       "  'rs63750053',\n",
       "  'rs63750082',\n",
       "  'rs63750083',\n",
       "  'rs63750218',\n",
       "  'rs63750219',\n",
       "  'rs63750227',\n",
       "  'rs63750231',\n",
       "  'rs63750248',\n",
       "  'rs63750265',\n",
       "  'rs63750301',\n",
       "  'rs63750306',\n",
       "  'rs63750311',\n",
       "  'rs63750322',\n",
       "  'rs63750323',\n",
       "  'rs63750353',\n",
       "  'rs63750391',\n",
       "  'rs63750418',\n",
       "  'rs63750450',\n",
       "  'rs63750470',\n",
       "  'rs63750526',\n",
       "  'rs63750550',\n",
       "  'rs63750577',\n",
       "  'rs63750590',\n",
       "  'rs63750599',\n",
       "  'rs63750631',\n",
       "  'rs63750646',\n",
       "  'rs63750687',\n",
       "  'rs63750730',\n",
       "  'rs63750761',\n",
       "  'rs63750779',\n",
       "  'rs63750800',\n",
       "  'rs63750858',\n",
       "  'rs63750886',\n",
       "  'rs63750900',\n",
       "  'rs63750929',\n",
       "  'rs63750963',\n",
       "  'rs63750964',\n",
       "  'rs63751003',\n",
       "  'rs63751024',\n",
       "  'rs63751037',\n",
       "  'rs63751068',\n",
       "  'rs63751106',\n",
       "  'rs63751139',\n",
       "  'rs63751141',\n",
       "  'rs63751144',\n",
       "  'rs63751163',\n",
       "  'rs63751210',\n",
       "  'rs63751223',\n",
       "  'rs63751229',\n",
       "  'rs63751235',\n",
       "  'rs63751272',\n",
       "  'rs63751278',\n",
       "  'rs63751287',\n",
       "  'rs63751316',\n",
       "  'rs63751399',\n",
       "  'rs63751416',\n",
       "  'rs63751475',\n",
       "  'rs267606983',\n",
       "  'rs281875357',\n",
       "  'rs778630379',\n",
       "  'rs764971634',\n",
       "  'rs1057518919',\n",
       "  'rs1362575880',\n",
       "  'rs1566630791',\n",
       "  'rs1566630811',\n",
       "  'rs1566630884',\n",
       "  'rs1566630910',\n",
       "  'rs1566638673',\n",
       "  'rs1566641934',\n",
       "  'rs1566650594',\n",
       "  'rs1566656702',\n",
       "  'rs1566657804',\n",
       "  'rs1555355250',\n",
       "  'rs1555355289',\n",
       "  'rs1555357544',\n",
       "  'rs1555358260',\n",
       "  'rs1594998354',\n",
       "  'rs1595035030',\n",
       "  'rs1897874329',\n",
       "  'rs1897876766',\n",
       "  'rs1897878139',\n",
       "  'rs1897961716',\n",
       "  'rs1898534725',\n",
       "  'rs1898781850',\n",
       "  'rs2140037322',\n",
       "  'rs2140080563',\n",
       "  'rs2140105309',\n",
       "  'rs2140145565'],\n",
       " 'ENSG00000131711': ['rs146324682',\n",
       "  'rs371409009',\n",
       "  'rs753026898',\n",
       "  'rs1554054827',\n",
       "  'rs1554054831',\n",
       "  'rs1554054880',\n",
       "  'rs1554055106',\n",
       "  'rs1561314246',\n",
       "  'rs1561315170',\n",
       "  'rs1580026238',\n",
       "  'rs1580027036',\n",
       "  'rs1747146910',\n",
       "  'rs1747274021',\n",
       "  'rs1747234497',\n",
       "  'rs2111893851',\n",
       "  'rs2111894733',\n",
       "  'rs2111894757',\n",
       "  'rs2112231793',\n",
       "  'rs2112238696'],\n",
       " 'ENSG00000136717': ['rs114833236',\n",
       "  'rs115938552',\n",
       "  'rs78967885',\n",
       "  'rs121909273',\n",
       "  'rs121909274',\n",
       "  'rs121909275',\n",
       "  'rs138047593',\n",
       "  'rs61748158',\n",
       "  'rs35535012',\n",
       "  'rs143258043',\n",
       "  'rs148945502',\n",
       "  'rs138606879',\n",
       "  'rs148179522',\n",
       "  'rs199658397',\n",
       "  'rs201397427',\n",
       "  'rs267606681',\n",
       "  'rs370911793',\n",
       "  'rs371258305',\n",
       "  'rs375583449',\n",
       "  'rs374490688',\n",
       "  'rs372360787',\n",
       "  'rs372449622',\n",
       "  'rs535176915',\n",
       "  'rs556129959',\n",
       "  'rs587783343',\n",
       "  'rs748026377',\n",
       "  'rs759676621',\n",
       "  'rs764951306',\n",
       "  'rs777176261',\n",
       "  'rs776737413',\n",
       "  'rs765870974',\n",
       "  'rs779756862',\n",
       "  'rs773313892',\n",
       "  'rs1249621033',\n",
       "  'rs1553458019',\n",
       "  'rs1682909859'],\n",
       " 'ENSG00000140459': ['rs6161',\n",
       "  'rs143655263',\n",
       "  'rs387906601',\n",
       "  'rs121912811',\n",
       "  'rs121912812',\n",
       "  'rs121912813',\n",
       "  'rs121912814',\n",
       "  'rs72547508',\n",
       "  'rs886041121',\n",
       "  'rs886041759',\n",
       "  'rs755186597',\n",
       "  'rs757299093',\n",
       "  'rs762412759',\n",
       "  'rs775102947',\n",
       "  'rs776056840',\n",
       "  'rs1021942880',\n",
       "  'rs1287871034',\n",
       "  'rs1567053134',\n",
       "  'rs2060619548',\n",
       "  'rs2060625814',\n",
       "  'rs2060627551',\n",
       "  'rs2060639501',\n",
       "  'rs6161',\n",
       "  'rs143655263',\n",
       "  'rs387906601',\n",
       "  'rs121912811',\n",
       "  'rs121912812',\n",
       "  'rs121912813',\n",
       "  'rs121912814',\n",
       "  'rs72547508',\n",
       "  'rs886041121',\n",
       "  'rs886041759',\n",
       "  'rs755186597',\n",
       "  'rs757299093',\n",
       "  'rs762412759',\n",
       "  'rs775102947',\n",
       "  'rs776056840',\n",
       "  'rs1021942880',\n",
       "  'rs1287871034',\n",
       "  'rs1567053134',\n",
       "  'rs2060619548',\n",
       "  'rs2060625814',\n",
       "  'rs2060627551',\n",
       "  'rs2060639501'],\n",
       " 'ENSG00000185760': ['rs530101249',\n",
       "  'rs549110435',\n",
       "  'rs754929718',\n",
       "  'rs756542782',\n",
       "  'rs773692396',\n",
       "  'rs776643693',\n",
       "  'rs1135401955',\n",
       "  'rs1135401956',\n",
       "  'rs1135401957',\n",
       "  'rs1135401958',\n",
       "  'rs1314919218',\n",
       "  'rs1554201137',\n",
       "  'rs1554210418',\n",
       "  'rs1554213240',\n",
       "  'rs1771729432',\n",
       "  'rs1775232605',\n",
       "  'rs1775233922',\n",
       "  'rs1776322078',\n",
       "  'rs2150357621',\n",
       "  'rs2150391860',\n",
       "  'rs1554210415'],\n",
       " 'ENSG00000046604': ['rs2230234',\n",
       "  'rs185821167',\n",
       "  'rs191300661',\n",
       "  'rs191564916',\n",
       "  'rs180695545',\n",
       "  'rs180926981',\n",
       "  'rs531036279',\n",
       "  'rs539821357',\n",
       "  'rs146402368',\n",
       "  'rs114544564',\n",
       "  'rs121913006',\n",
       "  'rs121913007',\n",
       "  'rs121913008',\n",
       "  'rs121913010',\n",
       "  'rs121913011',\n",
       "  'rs121913013',\n",
       "  'rs140575919',\n",
       "  'rs141405267',\n",
       "  'rs553299589',\n",
       "  'rs79068489',\n",
       "  'rs113451409',\n",
       "  'rs565904909',\n",
       "  'rs372174546',\n",
       "  'rs373542380',\n",
       "  'rs375527314',\n",
       "  'rs376424003',\n",
       "  'rs376556524',\n",
       "  'rs576404380',\n",
       "  'rs727502990',\n",
       "  'rs727504783',\n",
       "  'rs730880347',\n",
       "  'rs745457570',\n",
       "  'rs193298428',\n",
       "  'rs193922639',\n",
       "  'rs199681901',\n",
       "  'rs199761749',\n",
       "  'rs200201181',\n",
       "  'rs200509948',\n",
       "  'rs200997703',\n",
       "  'rs201051252',\n",
       "  'rs201855245',\n",
       "  'rs202057770',\n",
       "  'rs202063433',\n",
       "  'rs369489095',\n",
       "  'rs370509593',\n",
       "  'rs371146201',\n",
       "  'rs746353565',\n",
       "  'rs750176752',\n",
       "  'rs752432726',\n",
       "  'rs752522753',\n",
       "  'rs753052874',\n",
       "  'rs547406532',\n",
       "  'rs62095193',\n",
       "  'rs762526848',\n",
       "  'rs763907170',\n",
       "  'rs779360113',\n",
       "  'rs780469370',\n",
       "  'rs781532110',\n",
       "  'rs794728081',\n",
       "  'rs794728083',\n",
       "  'rs794728084',\n",
       "  'rs794728086',\n",
       "  'rs794728091',\n",
       "  'rs794728092',\n",
       "  'rs794728093',\n",
       "  'rs794728094',\n",
       "  'rs794728097',\n",
       "  'rs794728098',\n",
       "  'rs794728100',\n",
       "  'rs546127303',\n",
       "  'rs376573409',\n",
       "  'rs397514038',\n",
       "  'rs397516702',\n",
       "  'rs397516703',\n",
       "  'rs397516706',\n",
       "  'rs397516709',\n",
       "  'rs397516712',\n",
       "  'rs754133577',\n",
       "  'rs755243947',\n",
       "  'rs770921270',\n",
       "  'rs772744115',\n",
       "  'rs773328409',\n",
       "  'rs773812422',\n",
       "  'rs775256998',\n",
       "  'rs775642244',\n",
       "  'rs776078563',\n",
       "  'rs776516070',\n",
       "  'rs776975980',\n",
       "  'rs778100657',\n",
       "  'rs758282201',\n",
       "  'rs758537946',\n",
       "  'rs759944835',\n",
       "  'rs869025388',\n",
       "  'rs876657791',\n",
       "  'rs1021457619',\n",
       "  'rs1039633976',\n",
       "  'rs1064793103',\n",
       "  'rs1064793983',\n",
       "  'rs1064794709',\n",
       "  'rs1158782181',\n",
       "  'rs1187924885',\n",
       "  'rs1165139589',\n",
       "  'rs1237620145',\n",
       "  'rs1250563249',\n",
       "  'rs1261674855',\n",
       "  'rs1257877047',\n",
       "  'rs1284598654',\n",
       "  'rs1252006527',\n",
       "  'rs1252426323',\n",
       "  'rs1342580910',\n",
       "  'rs1334657349',\n",
       "  'rs1375012922',\n",
       "  'rs1375081885',\n",
       "  'rs1389484271',\n",
       "  'rs1382430464',\n",
       "  'rs1316380114',\n",
       "  'rs1452179158',\n",
       "  'rs1472037685',\n",
       "  'rs1567933176',\n",
       "  'rs1567934720',\n",
       "  'rs1567934773',\n",
       "  'rs1568098570',\n",
       "  'rs1568105371',\n",
       "  'rs1598810829',\n",
       "  'rs1598811348',\n",
       "  'rs1598811394',\n",
       "  'rs1598826658',\n",
       "  'rs1555627108',\n",
       "  'rs1555627118',\n",
       "  'rs1555671201',\n",
       "  'rs1555671331',\n",
       "  'rs1555671335',\n",
       "  'rs1555671441',\n",
       "  'rs2073125384',\n",
       "  'rs2073148989',\n",
       "  'rs2073149649',\n",
       "  'rs2073222422',\n",
       "  'rs2073265410',\n",
       "  'rs2073308127',\n",
       "  'rs2073310848',\n",
       "  'rs2144315483',\n",
       "  'rs2144316016',\n",
       "  'rs2144317638',\n",
       "  'rs2144317658',\n",
       "  'rs2144317934',\n",
       "  'rs2144318119',\n",
       "  'rs2144322548',\n",
       "  'rs2144322985',\n",
       "  'rs2144323019',\n",
       "  'rs2144332473',\n",
       "  'rs2144332585',\n",
       "  'rs2144332631',\n",
       "  'rs2144339598',\n",
       "  'rs2144339967',\n",
       "  'rs2144346522'],\n",
       " 'ENSG00000064012': ['rs17860424',\n",
       "  'rs143410219',\n",
       "  'rs146286958',\n",
       "  'rs375465590',\n",
       "  'rs373232880',\n",
       "  'rs587776665',\n",
       "  'rs747862347',\n",
       "  'rs765672355',\n",
       "  'rs762135470',\n",
       "  'rs768360449',\n",
       "  'rs777463099',\n",
       "  'rs777784105',\n",
       "  'rs776712453',\n",
       "  'rs1368296717',\n",
       "  'rs1457092267',\n",
       "  'rs1559374786',\n",
       "  'rs1948252170'],\n",
       " 'ENSG00000086288': ['rs121918300', 'rs146777129'],\n",
       " 'ENSG00000106460': ['rs1554310600'],\n",
       " 'ENSG00000145979': ['rs112668572',\n",
       "  'rs483352922',\n",
       "  'rs758759342',\n",
       "  'rs587777652',\n",
       "  'rs1554192525',\n",
       "  'rs1584522453'],\n",
       " 'ENSG00000152223': ['rs190673127',\n",
       "  'rs191244915',\n",
       "  'rs183478189',\n",
       "  'rs144334723',\n",
       "  'rs116076204',\n",
       "  'rs552419530',\n",
       "  'rs148641800',\n",
       "  'rs562098384',\n",
       "  'rs372940918',\n",
       "  'rs373944025',\n",
       "  'rs374321183',\n",
       "  'rs376297358',\n",
       "  'rs587776939',\n",
       "  'rs587776940',\n",
       "  'rs587776941',\n",
       "  'rs587776942',\n",
       "  'rs201067154',\n",
       "  'rs201757275',\n",
       "  'rs370720753',\n",
       "  'rs746036349',\n",
       "  'rs746885334',\n",
       "  'rs748670315',\n",
       "  'rs762639913',\n",
       "  'rs763614919',\n",
       "  'rs763788808',\n",
       "  'rs763931697',\n",
       "  'rs780889226',\n",
       "  'rs863225064',\n",
       "  'rs886043244',\n",
       "  'rs886276412',\n",
       "  'rs755928939',\n",
       "  'rs756503608',\n",
       "  'rs912986968',\n",
       "  'rs772325682',\n",
       "  'rs773330060',\n",
       "  'rs776146922',\n",
       "  'rs767638289',\n",
       "  'rs769440447',\n",
       "  'rs769833086',\n",
       "  'rs908790331',\n",
       "  'rs759625169',\n",
       "  'rs759930622',\n",
       "  'rs760768451',\n",
       "  'rs761554022',\n",
       "  'rs866435487',\n",
       "  'rs961245497',\n",
       "  'rs986592823',\n",
       "  'rs1057519318',\n",
       "  'rs1064795230',\n",
       "  'rs1064795606',\n",
       "  'rs1085308062',\n",
       "  'rs1135402727',\n",
       "  'rs1135402728',\n",
       "  'rs1135402729',\n",
       "  'rs1135402730',\n",
       "  'rs1135402731',\n",
       "  'rs1135402732',\n",
       "  'rs1135402733',\n",
       "  'rs1135402734',\n",
       "  'rs1135402735',\n",
       "  'rs1135402736',\n",
       "  'rs1161436837',\n",
       "  'rs1203870830',\n",
       "  'rs1470797555',\n",
       "  'rs1568103027',\n",
       "  'rs1568107449',\n",
       "  'rs1568112516',\n",
       "  'rs1568112543',\n",
       "  'rs1568118775',\n",
       "  'rs1568133724',\n",
       "  'rs1568133760',\n",
       "  'rs1568142333',\n",
       "  'rs1568150793',\n",
       "  'rs1568168835',\n",
       "  'rs1568183567',\n",
       "  'rs1555667231',\n",
       "  'rs1555668791',\n",
       "  'rs1555671254',\n",
       "  'rs1555673890',\n",
       "  'rs1555673917',\n",
       "  'rs1555681647',\n",
       "  'rs1479239564',\n",
       "  'rs1483969601',\n",
       "  'rs1599445663',\n",
       "  'rs1599447883',\n",
       "  'rs1599474108',\n",
       "  'rs1599536432',\n",
       "  'rs1599578081',\n",
       "  'rs2048968935',\n",
       "  'rs2049058963',\n",
       "  'rs2049301523',\n",
       "  'rs2050937885',\n",
       "  'rs2050008898',\n",
       "  'rs2050027504',\n",
       "  'rs2050481336',\n",
       "  'rs2050617604',\n",
       "  'rs2050902019',\n",
       "  'rs2145173322',\n",
       "  'rs2145393546',\n",
       "  'rs2145422618',\n",
       "  'rs2145667073',\n",
       "  'rs2145712492',\n",
       "  'rs2145713137',\n",
       "  'rs2145807103',\n",
       "  'rs2145880933'],\n",
       " 'ENSG00000154734': ['rs375553171'],\n",
       " 'ENSG00000157954': ['rs756429763', 'rs2115280647'],\n",
       " 'ENSG00000167397': ['rs104894539',\n",
       "  'rs104894540',\n",
       "  'rs104894541',\n",
       "  'rs9934438',\n",
       "  'rs61742245',\n",
       "  'rs72547528',\n",
       "  'rs72547529'],\n",
       " 'ENSG00000168769': ['rs116519313',\n",
       "  'rs377382567',\n",
       "  'rs576934285',\n",
       "  'rs759055581',\n",
       "  'rs1131691943',\n",
       "  'rs1231721810',\n",
       "  'rs1267284740',\n",
       "  'rs1283023309',\n",
       "  'rs1578673280',\n",
       "  'rs2110231198'],\n",
       " 'ENSG00000169836': ['rs121918124',\n",
       "  'rs121918125',\n",
       "  'rs144292455',\n",
       "  'rs146482011',\n",
       "  'rs201195175',\n",
       "  'rs397515483',\n",
       "  'rs527698386',\n",
       "  'rs727505375',\n",
       "  'rs760022956',\n",
       "  'rs764659822',\n",
       "  'rs772508077',\n",
       "  'rs1578272223'],\n",
       " 'ENSG00000221838': ['rs138437966',\n",
       "  'rs144875978',\n",
       "  'rs139861201',\n",
       "  'rs147738731',\n",
       "  'rs146262009',\n",
       "  'rs141754568',\n",
       "  'rs369459721',\n",
       "  'rs371712994',\n",
       "  'rs387906838',\n",
       "  'rs531148057',\n",
       "  'rs372641895',\n",
       "  'rs542768500',\n",
       "  'rs574878636',\n",
       "  'rs730882249',\n",
       "  'rs752598529',\n",
       "  'rs750783717',\n",
       "  'rs755533568',\n",
       "  'rs754498075',\n",
       "  'rs760907496',\n",
       "  'rs758387391',\n",
       "  'rs764326593',\n",
       "  'rs780030221',\n",
       "  'rs770705832',\n",
       "  'rs772119268',\n",
       "  'rs797045249',\n",
       "  'rs769952838',\n",
       "  'rs886041126',\n",
       "  'rs776788025',\n",
       "  'rs777220438',\n",
       "  'rs766513025',\n",
       "  'rs973744863',\n",
       "  'rs1057521784',\n",
       "  'rs1064794383',\n",
       "  'rs1131691556',\n",
       "  'rs1191415115',\n",
       "  'rs1293317548',\n",
       "  'rs1344550427',\n",
       "  'rs1321353475',\n",
       "  'rs1350685858',\n",
       "  'rs1397027035',\n",
       "  'rs1432802140',\n",
       "  'rs1437308674',\n",
       "  'rs1437455190',\n",
       "  'rs1554377262',\n",
       "  'rs1554380093',\n",
       "  'rs1554380391',\n",
       "  'rs1562912305',\n",
       "  'rs1584507780',\n",
       "  'rs1584508277',\n",
       "  'rs1584514057',\n",
       "  'rs1796176171',\n",
       "  'rs1796279189',\n",
       "  'rs1796054104',\n",
       "  'rs2116614049',\n",
       "  'rs2116616864',\n",
       "  'rs2116628120',\n",
       "  'rs2116638409',\n",
       "  'rs2116643498',\n",
       "  'rs2116668499',\n",
       "  'rs2116670931'],\n",
       " 'ENSG00000102935': ['rs548986682',\n",
       "  'rs745597535',\n",
       "  'rs886041561',\n",
       "  'rs1596759273',\n",
       "  'rs1596763347',\n",
       "  'rs1596988259'],\n",
       " 'ENSG00000104783': ['rs76935412',\n",
       "  'rs774455945',\n",
       "  'rs1057519076',\n",
       "  'rs1057519077',\n",
       "  'rs1969582489'],\n",
       " 'ENSG00000107736': ['rs115398922',\n",
       "  'rs185105210',\n",
       "  'rs79636933',\n",
       "  'rs111033247',\n",
       "  'rs111033270',\n",
       "  'rs111033271',\n",
       "  'rs111033369',\n",
       "  'rs111033457',\n",
       "  'rs111033458',\n",
       "  'rs111033473',\n",
       "  'rs111033487',\n",
       "  'rs111033490',\n",
       "  'rs111033493',\n",
       "  'rs111033522',\n",
       "  'rs111033536',\n",
       "  'rs181255269',\n",
       "  'rs181611778',\n",
       "  'rs181658753',\n",
       "  'rs188098974',\n",
       "  'rs188376296',\n",
       "  'rs191021194',\n",
       "  'rs191154178',\n",
       "  'rs183431253',\n",
       "  'rs142788731',\n",
       "  'rs142857685',\n",
       "  'rs144688588',\n",
       "  'rs71012280',\n",
       "  'rs528880720',\n",
       "  'rs121908347',\n",
       "  'rs121908348',\n",
       "  'rs121908349',\n",
       "  'rs121908350',\n",
       "  'rs121908351',\n",
       "  'rs121908352',\n",
       "  'rs121908353',\n",
       "  'rs121908354',\n",
       "  'rs121908355',\n",
       "  'rs139287714',\n",
       "  'rs139409005',\n",
       "  'rs537236734',\n",
       "  'rs140884994',\n",
       "  'rs142131750',\n",
       "  'rs376271562',\n",
       "  'rs376332131',\n",
       "  'rs376560330',\n",
       "  'rs376881824',\n",
       "  'rs377614198',\n",
       "  'rs397517305',\n",
       "  'rs397517307',\n",
       "  'rs397517313',\n",
       "  'rs397517323',\n",
       "  'rs397517324',\n",
       "  'rs397517326',\n",
       "  'rs397517327',\n",
       "  'rs397517328',\n",
       "  'rs397517329',\n",
       "  'rs397517331',\n",
       "  'rs397517333',\n",
       "  'rs397517334',\n",
       "  'rs397517337',\n",
       "  'rs397517341',\n",
       "  'rs397517342',\n",
       "  'rs397517344',\n",
       "  'rs397517346',\n",
       "  'rs397517349',\n",
       "  'rs397517350',\n",
       "  'rs397517353',\n",
       "  'rs397517354',\n",
       "  'rs397517356',\n",
       "  'rs397517362',\n",
       "  'rs397517367',\n",
       "  'rs368828743',\n",
       "  'rs368841307',\n",
       "  'rs369013539',\n",
       "  'rs369396703',\n",
       "  'rs369501114',\n",
       "  'rs369697366',\n",
       "  'rs369900526',\n",
       "  'rs370074117',\n",
       "  'rs370568585',\n",
       "  'rs370762269',\n",
       "  'rs370782827',\n",
       "  'rs547034667',\n",
       "  'rs531513127',\n",
       "  'rs192326086',\n",
       "  'rs199508694',\n",
       "  'rs199510686',\n",
       "  'rs199894395',\n",
       "  'rs199903227',\n",
       "  'rs200251748',\n",
       "  'rs200324241',\n",
       "  'rs200542052',\n",
       "  'rs200632520',\n",
       "  'rs200635365',\n",
       "  'rs200638595',\n",
       "  'rs200649500',\n",
       "  'rs186394654',\n",
       "  'rs186866326',\n",
       "  'rs186990940',\n",
       "  'rs550384315',\n",
       "  'rs370983472',\n",
       "  'rs371522435',\n",
       "  'rs371609236',\n",
       "  'rs371910002',\n",
       "  'rs371932558',\n",
       "  'rs372158876',\n",
       "  'rs372172457',\n",
       "  'rs372388344',\n",
       "  'rs372588663',\n",
       "  'rs373269394',\n",
       "  'rs201024982',\n",
       "  'rs201053044',\n",
       "  'rs201076440',\n",
       "  'rs201232514',\n",
       "  'rs201297042',\n",
       "  'rs201364852',\n",
       "  'rs201434373',\n",
       "  'rs201475055',\n",
       "  'rs202052174',\n",
       "  'rs202101019',\n",
       "  'rs202147495',\n",
       "  'rs367750869',\n",
       "  'rs367928692',\n",
       "  'rs45522532',\n",
       "  'rs55947063',\n",
       "  'rs56107171',\n",
       "  'rs34432586',\n",
       "  'rs41281336',\n",
       "  'rs41281338',\n",
       "  'rs540567272',\n",
       "  'rs554353516',\n",
       "  'rs554938323',\n",
       "  'rs555430482',\n",
       "  'rs557620034',\n",
       "  'rs552998089',\n",
       "  'rs562525508',\n",
       "  'rs562590210',\n",
       "  'rs542618168',\n",
       "  'rs558564568',\n",
       "  'rs560760228',\n",
       "  'rs373631099',\n",
       "  'rs373838930',\n",
       "  'rs374156784',\n",
       "  'rs374215303',\n",
       "  'rs374362883',\n",
       "  'rs374545987',\n",
       "  'rs374549358',\n",
       "  'rs375292899',\n",
       "  'rs375358318',\n",
       "  'rs375781856',\n",
       "  'rs727502919',\n",
       "  'rs727502931',\n",
       "  'rs727502933',\n",
       "  'rs727503841',\n",
       "  'rs727504455',\n",
       "  'rs727504715',\n",
       "  'rs727504761',\n",
       "  'rs745571683',\n",
       "  'rs745668474',\n",
       "  'rs753886326',\n",
       "  'rs754876029',\n",
       "  'rs759981467',\n",
       "  'rs761913744',\n",
       "  'rs762118583',\n",
       "  'rs746152246',\n",
       "  'rs747955135',\n",
       "  'rs573057228',\n",
       "  'rs764824311',\n",
       "  'rs764949139',\n",
       "  'rs765026901',\n",
       "  'rs766541944',\n",
       "  'rs769433759',\n",
       "  'rs769524849',\n",
       "  'rs769742202',\n",
       "  'rs769870573',\n",
       "  'rs769879897',\n",
       "  'rs769972347',\n",
       "  'rs770665588',\n",
       "  'rs771002870',\n",
       "  'rs771210121',\n",
       "  'rs771353319',\n",
       "  'rs750803248',\n",
       "  'rs750880909',\n",
       "  'rs751788879',\n",
       "  'rs757675676',\n",
       "  'rs758382198',\n",
       "  'rs780987516',\n",
       "  'rs786205612',\n",
       "  'rs794727649',\n",
       "  'rs796051860',\n",
       "  'rs796051861',\n",
       "  'rs766906353',\n",
       "  'rs767004225',\n",
       "  'rs767343063',\n",
       "  'rs768834772',\n",
       "  'rs776354402',\n",
       "  'rs776501112',\n",
       "  'rs778251205',\n",
       "  'rs866199775',\n",
       "  'rs866435331',\n",
       "  'rs771555570',\n",
       "  'rs771766431',\n",
       "  'rs772953914',\n",
       "  'rs773004408',\n",
       "  'rs773464867',\n",
       "  'rs889241219',\n",
       "  'rs779038178',\n",
       "  'rs779425775',\n",
       "  'rs779974496',\n",
       "  'rs780917129',\n",
       "  'rs868188730',\n",
       "  'rs876657636',\n",
       "  'rs876657680',\n",
       "  'rs876657682',\n",
       "  'rs876657754',\n",
       "  'rs876657756',\n",
       "  'rs878853337',\n",
       "  'rs886038668',\n",
       "  'rs756147087',\n",
       "  'rs756336099',\n",
       "  'rs756454038',\n",
       "  'rs756674688',\n",
       "  'rs757053706',\n",
       "  'rs773926246',\n",
       "  'rs774559018',\n",
       "  'rs775093336',\n",
       "  'rs748828988',\n",
       "  'rs749203752',\n",
       "  'rs749235545',\n",
       "  'rs749236786',\n",
       "  'rs749287212',\n",
       "  'rs750027965',\n",
       "  'rs936479651',\n",
       "  'rs762226905',\n",
       "  'rs762805265',\n",
       "  'rs763595608',\n",
       "  'rs763721044',\n",
       "  'rs763820312',\n",
       "  'rs764025875',\n",
       "  'rs986154966',\n",
       "  'rs1039517349',\n",
       "  'rs1057519500',\n",
       "  'rs1057520661',\n",
       "  'rs1057520662',\n",
       "  'rs1057524265',\n",
       "  'rs1060499714',\n",
       "  'rs1060499788',\n",
       "  'rs1060499789',\n",
       "  'rs1060499790',\n",
       "  'rs1060499791',\n",
       "  'rs1060499792',\n",
       "  'rs1060499793',\n",
       "  'rs1064795044',\n",
       "  'rs1064795722',\n",
       "  'rs1052484950',\n",
       "  'rs1166948274',\n",
       "  'rs1184309321',\n",
       "  'rs1190307769',\n",
       "  'rs1204500829',\n",
       "  'rs1200012430',\n",
       "  'rs1216140697',\n",
       "  'rs1230303971',\n",
       "  'rs1253419936',\n",
       "  'rs1221464948',\n",
       "  'rs1228413559',\n",
       "  'rs1270566026',\n",
       "  'rs1260523958',\n",
       "  'rs1264310782',\n",
       "  'rs1274105133',\n",
       "  'rs1278603247',\n",
       "  'rs1280991447',\n",
       "  'rs1293118415',\n",
       "  'rs1306728898',\n",
       "  'rs1292050472',\n",
       "  'rs1347367555',\n",
       "  'rs1338446830',\n",
       "  'rs1346156899',\n",
       "  'rs1344509500',\n",
       "  'rs1362644811',\n",
       "  'rs1377982927',\n",
       "  'rs1379171478',\n",
       "  'rs1390562340',\n",
       "  'rs1385831846',\n",
       "  'rs1393567447',\n",
       "  'rs1416386069',\n",
       "  'rs1375225680',\n",
       "  'rs1375264465',\n",
       "  'rs1431027601',\n",
       "  'rs1397549896',\n",
       "  'rs1470508986',\n",
       "  'rs1474524543',\n",
       "  'rs1564623280',\n",
       "  'rs1487026359',\n",
       "  'rs1554856042',\n",
       "  'rs1554857840',\n",
       "  'rs1554858698',\n",
       "  'rs1554862262',\n",
       "  'rs1554871816',\n",
       "  'rs1554874373',\n",
       "  'rs1554874879',\n",
       "  'rs1554874884',\n",
       "  'rs1554874900',\n",
       "  'rs1554876990',\n",
       "  'rs1554877007',\n",
       "  'rs1554877797',\n",
       "  'rs1554877806',\n",
       "  'rs1564749141',\n",
       "  'rs1564759522',\n",
       "  'rs1564759653',\n",
       "  'rs1564788732',\n",
       "  'rs1564791773',\n",
       "  'rs1564794944',\n",
       "  'rs1564795354',\n",
       "  'rs1564796487',\n",
       "  'rs1564796673',\n",
       "  'rs1564803868',\n",
       "  'rs1564804195',\n",
       "  'rs1564805114',\n",
       "  'rs1564807887',\n",
       "  'rs1564808024',\n",
       "  'rs1589344004',\n",
       "  'rs1589384283',\n",
       "  'rs1589384720',\n",
       "  'rs1589292855',\n",
       "  'rs1589420011',\n",
       "  'rs1589424694',\n",
       "  'rs1589438942',\n",
       "  'rs1839422034',\n",
       "  'rs1839681191',\n",
       "  'rs1839717773',\n",
       "  'rs1840892582',\n",
       "  'rs1841212177',\n",
       "  'rs1841308758',\n",
       "  'rs1841311708',\n",
       "  'rs1841315122',\n",
       "  'rs1841599143',\n",
       "  'rs1841616050',\n",
       "  'rs1841921374',\n",
       "  'rs1841926049',\n",
       "  'rs1841971727',\n",
       "  'rs1841976391',\n",
       "  'rs1841984743',\n",
       "  'rs1841984906',\n",
       "  'rs1841885880',\n",
       "  'rs1841758961',\n",
       "  'rs1841773052',\n",
       "  'rs1841460358',\n",
       "  'rs1841497902',\n",
       "  'rs1857435780',\n",
       "  'rs1857438656',\n",
       "  'rs1857745217',\n",
       "  'rs1857755312',\n",
       "  'rs1850169561',\n",
       "  'rs1861130719',\n",
       "  'rs1861132418',\n",
       "  'rs1861138703',\n",
       "  'rs1861245902',\n",
       "  'rs1862873659',\n",
       "  'rs1862888748',\n",
       "  'rs1864433756',\n",
       "  'rs1864695566',\n",
       "  'rs1864965277',\n",
       "  'rs1864965345',\n",
       "  'rs1865145294',\n",
       "  'rs1865149889',\n",
       "  'rs1865609793',\n",
       "  'rs1865634253',\n",
       "  'rs1865727404',\n",
       "  'rs1865731567',\n",
       "  'rs1865887896',\n",
       "  'rs1866696630',\n",
       "  'rs1853882557',\n",
       "  'rs1853943430',\n",
       "  'rs2132400338',\n",
       "  'rs2132358279',\n",
       "  'rs2132358732',\n",
       "  'rs2132739739',\n",
       "  'rs2132740161',\n",
       "  'rs2132740263',\n",
       "  'rs2132745856',\n",
       "  'rs2132751512',\n",
       "  'rs2132751525',\n",
       "  'rs2132760607',\n",
       "  'rs2132800019',\n",
       "  'rs2132800290',\n",
       "  'rs2132804872',\n",
       "  'rs2132804878',\n",
       "  'rs2132805056',\n",
       "  'rs2132805287',\n",
       "  'rs2132820242',\n",
       "  'rs2132820264',\n",
       "  'rs2132820475',\n",
       "  'rs2132826354',\n",
       "  'rs2132841425',\n",
       "  'rs2132841517',\n",
       "  'rs2132846050',\n",
       "  'rs2132846322',\n",
       "  'rs2132195736',\n",
       "  'rs2132195841',\n",
       "  'rs2132196113',\n",
       "  'rs2132200416',\n",
       "  'rs2132664539',\n",
       "  'rs2132669396',\n",
       "  'rs2132682822',\n",
       "  'rs2132732963',\n",
       "  'rs2132733369',\n",
       "  'rs2132733452',\n",
       "  'rs2132023620',\n",
       "  'rs2132978502',\n",
       "  'rs2132978731',\n",
       "  'rs2132983525',\n",
       "  'rs2132983724',\n",
       "  'rs2132984089',\n",
       "  'rs2132985083',\n",
       "  'rs2132987509',\n",
       "  'rs2132988621',\n",
       "  'rs2132988732',\n",
       "  'rs2132990173',\n",
       "  'rs2132990504',\n",
       "  'rs2132990828',\n",
       "  'rs2132996862',\n",
       "  'rs2133001560',\n",
       "  'rs2133002032',\n",
       "  'rs2133002381',\n",
       "  'rs2133002929',\n",
       "  'rs2133003011',\n",
       "  'rs2133003145',\n",
       "  'rs2133006702',\n",
       "  'rs2133007155',\n",
       "  'rs2132925663',\n",
       "  'rs2132925769',\n",
       "  'rs2132927249',\n",
       "  'rs2132927381',\n",
       "  'rs2132929231',\n",
       "  'rs2132929303',\n",
       "  'rs2132929769',\n",
       "  'rs2132929805',\n",
       "  'rs2132930073',\n",
       "  'rs2132938020',\n",
       "  'rs2132938148',\n",
       "  'rs2132939079',\n",
       "  'rs2132940635',\n",
       "  'rs2132945702',\n",
       "  'rs2132945762',\n",
       "  'rs2132945765',\n",
       "  'rs2132947899',\n",
       "  'rs2132949774',\n",
       "  'rs2132949791',\n",
       "  'rs2132953500',\n",
       "  'rs2132953942',\n",
       "  'rs2132954257',\n",
       "  'rs2132965694',\n",
       "  'rs2132965891',\n",
       "  'rs2132967710',\n",
       "  'rs2132967767',\n",
       "  'rs2132967790',\n",
       "  'rs2132968121',\n",
       "  'rs2132968572',\n",
       "  'rs2132968839',\n",
       "  'rs2132971828',\n",
       "  'rs2132514996',\n",
       "  'rs2132594619',\n",
       "  'rs2132594712',\n",
       "  'rs2132596686'],\n",
       " 'ENSG00000130203': ['rs7412',\n",
       "  'rs429358',\n",
       "  'rs769452',\n",
       "  'rs769455',\n",
       "  'rs190853081',\n",
       "  'rs121918392',\n",
       "  'rs121918393',\n",
       "  'rs121918394',\n",
       "  'rs121918395',\n",
       "  'rs121918396',\n",
       "  'rs121918397',\n",
       "  'rs121918398',\n",
       "  'rs121918399',\n",
       "  'rs387906567',\n",
       "  'rs397514253',\n",
       "  'rs397514254',\n",
       "  'rs515726148',\n",
       "  'rs140808909',\n",
       "  'rs201672011',\n",
       "  'rs267606661',\n",
       "  'rs267606662',\n",
       "  'rs267606663',\n",
       "  'rs267606664',\n",
       "  'rs199768005',\n",
       "  'rs200703101',\n",
       "  'rs28931576',\n",
       "  'rs28931577',\n",
       "  'rs28931578',\n",
       "  'rs28931579',\n",
       "  'rs1332591068',\n",
       "  'rs2122132718',\n",
       "  'rs2122137937'],\n",
       " 'ENSG00000163666': ['rs104893742',\n",
       "  'rs28936416',\n",
       "  'rs28936702',\n",
       "  'rs28936703',\n",
       "  'rs28936704',\n",
       "  'rs121909173',\n",
       "  'rs141270468',\n",
       "  'rs143057250',\n",
       "  'rs149663188',\n",
       "  'rs575112817',\n",
       "  'rs748460226',\n",
       "  'rs587776664',\n",
       "  'rs752434998',\n",
       "  'rs754137696',\n",
       "  'rs751011809',\n",
       "  'rs766234350',\n",
       "  'rs770886420',\n",
       "  'rs777223697',\n",
       "  'rs777833871',\n",
       "  'rs1238248024',\n",
       "  'rs1286347776',\n",
       "  'rs2060462273',\n",
       "  'rs2060466023',\n",
       "  'rs2107565760',\n",
       "  'rs2107566038'],\n",
       " 'ENSG00000189114': ['rs281865115',\n",
       "  'rs281865116',\n",
       "  'rs754841982',\n",
       "  'rs749030472',\n",
       "  'rs1568469902',\n",
       "  'rs1969482515'],\n",
       " 'ENSG00000197943': ['rs114618894',\n",
       "  'rs187956469',\n",
       "  'rs189301790',\n",
       "  'rs397514562',\n",
       "  'rs753618006',\n",
       "  'rs769393894',\n",
       "  'rs372557475',\n",
       "  'rs1057519831',\n",
       "  'rs1057519832',\n",
       "  'rs2143682648',\n",
       "  'rs2143705605'],\n",
       " 'ENSG00000108556': ['rs121909510',\n",
       "  'rs121909511',\n",
       "  'rs121909512',\n",
       "  'rs121909513',\n",
       "  'rs121909514',\n",
       "  'rs121909515',\n",
       "  'rs121909516',\n",
       "  'rs121909517',\n",
       "  'rs144169073',\n",
       "  'rs148370803',\n",
       "  'rs193919341',\n",
       "  'rs140023380',\n",
       "  'rs370019023',\n",
       "  'rs372635387',\n",
       "  'rs373710822',\n",
       "  'rs398122830',\n",
       "  'rs483353045',\n",
       "  'rs746199600',\n",
       "  'rs746993242',\n",
       "  'rs748103983',\n",
       "  'rs748289906',\n",
       "  'rs748694848',\n",
       "  'rs756675414',\n",
       "  'rs757396750',\n",
       "  'rs758687208',\n",
       "  'rs762368691',\n",
       "  'rs762704832',\n",
       "  'rs763258280',\n",
       "  'rs763943642',\n",
       "  'rs28929768',\n",
       "  'rs28999110',\n",
       "  'rs56377005',\n",
       "  'rs768552387',\n",
       "  'rs768869197',\n",
       "  'rs769945146',\n",
       "  'rs773916039',\n",
       "  'rs773929089',\n",
       "  'rs774425374',\n",
       "  'rs879253722',\n",
       "  'rs879253723',\n",
       "  'rs879255562',\n",
       "  'rs879255563',\n",
       "  'rs886037628',\n",
       "  'rs886043239',\n",
       "  'rs759226183',\n",
       "  'rs760022829',\n",
       "  'rs760565441',\n",
       "  'rs760623071',\n",
       "  'rs776927709',\n",
       "  'rs779816027',\n",
       "  'rs894382905',\n",
       "  'rs753828284',\n",
       "  'rs754975887',\n",
       "  'rs755303686',\n",
       "  'rs918839874',\n",
       "  'rs919271118',\n",
       "  'rs781774131',\n",
       "  'rs786204773',\n",
       "  'rs932032926',\n",
       "  'rs771486772',\n",
       "  'rs773526895',\n",
       "  'rs977512223',\n",
       "  'rs979787967',\n",
       "  'rs1156634884',\n",
       "  'rs1194524772',\n",
       "  'rs1168592296',\n",
       "  'rs1222065184',\n",
       "  'rs1255916068',\n",
       "  'rs1187421976',\n",
       "  'rs1268475945',\n",
       "  'rs1250853600',\n",
       "  'rs1208462125',\n",
       "  'rs1309292778',\n",
       "  'rs1320610655',\n",
       "  'rs1278514080',\n",
       "  'rs1407243713',\n",
       "  'rs1423995073',\n",
       "  'rs1396286715',\n",
       "  'rs1429181185',\n",
       "  'rs1430654625',\n",
       "  'rs1451776623',\n",
       "  'rs1445801336',\n",
       "  'rs1458613529',\n",
       "  'rs1555546038',\n",
       "  'rs1555546096',\n",
       "  'rs1555546224',\n",
       "  'rs1555546315',\n",
       "  'rs1555546465',\n",
       "  'rs1555546765',\n",
       "  'rs1555546858',\n",
       "  'rs1555547003',\n",
       "  'rs1398583523',\n",
       "  'rs1567635867',\n",
       "  'rs1567635954',\n",
       "  'rs1567636493',\n",
       "  'rs1567636622',\n",
       "  'rs1567638401',\n",
       "  'rs1567638550',\n",
       "  'rs1567638676',\n",
       "  'rs1597612665',\n",
       "  'rs1597613279',\n",
       "  'rs1597613302',\n",
       "  'rs1597613479',\n",
       "  'rs1597618787',\n",
       "  'rs1597618794',\n",
       "  'rs1597618854',\n",
       "  'rs1597619440',\n",
       "  'rs1597620368',\n",
       "  'rs1597621353',\n",
       "  'rs1597621396',\n",
       "  'rs1597622118',\n",
       "  'rs1969817843',\n",
       "  'rs1969834618',\n",
       "  'rs1969839736',\n",
       "  'rs1969856640',\n",
       "  'rs1969858094',\n",
       "  'rs1969880683',\n",
       "  'rs1969955431',\n",
       "  'rs1969972830',\n",
       "  'rs1969973509',\n",
       "  'rs1970005446',\n",
       "  'rs1970029155',\n",
       "  'rs1970039895',\n",
       "  'rs2151093952',\n",
       "  'rs2151094080',\n",
       "  'rs2151094378',\n",
       "  'rs2151094398',\n",
       "  'rs2151094540',\n",
       "  'rs2151094632',\n",
       "  'rs2151094729',\n",
       "  'rs2151095155',\n",
       "  'rs2151096983',\n",
       "  'rs2151097082',\n",
       "  'rs2151097188',\n",
       "  'rs2151097209',\n",
       "  'rs2151097420',\n",
       "  'rs2151097742',\n",
       "  'rs2151098283',\n",
       "  'rs2151098410',\n",
       "  'rs2151098412',\n",
       "  'rs2151098538',\n",
       "  'rs2151098864',\n",
       "  'rs2151098905',\n",
       "  'rs2151098928'],\n",
       " 'ENSG00000119684': ['rs565238806',\n",
       "  'rs138974583',\n",
       "  'rs181990057',\n",
       "  'rs142124529',\n",
       "  'rs142179277',\n",
       "  'rs797045117',\n",
       "  'rs144707485',\n",
       "  'rs148409389',\n",
       "  'rs376981545',\n",
       "  'rs61752722',\n",
       "  'rs370951929',\n",
       "  'rs775277584',\n",
       "  'rs776639203',\n",
       "  'rs772010423',\n",
       "  'rs762453215',\n",
       "  'rs587776622',\n",
       "  'rs745657036',\n",
       "  'rs746155226',\n",
       "  'rs746725349',\n",
       "  'rs28756978',\n",
       "  'rs28756987',\n",
       "  'rs28756990',\n",
       "  'rs28757008',\n",
       "  'rs28757011',\n",
       "  'rs28939071',\n",
       "  'rs41555714',\n",
       "  'rs764724825',\n",
       "  'rs1458734302',\n",
       "  'rs1431264077',\n",
       "  'rs1555343301'],\n",
       " 'ENSG00000136235': ['rs140352180',\n",
       "  'rs373987305',\n",
       "  'rs547758286',\n",
       "  'rs761169411',\n",
       "  'rs747723062',\n",
       "  'rs763065333',\n",
       "  'rs758729806',\n",
       "  'rs773435101',\n",
       "  'rs770211260',\n",
       "  'rs777204409',\n",
       "  'rs1554300664'],\n",
       " 'ENSG00000137642': ['rs142884576',\n",
       "  'rs201304369',\n",
       "  'rs770704659',\n",
       "  'rs758413354',\n",
       "  'rs1050845490',\n",
       "  'rs1229933491',\n",
       "  'rs2134823550',\n",
       "  'rs2134823666',\n",
       "  'rs2134893308'],\n",
       " 'ENSG00000139531': ['rs117778870',\n",
       "  'rs121908007',\n",
       "  'rs121908009',\n",
       "  'rs144064367',\n",
       "  'rs144469629',\n",
       "  'rs139444405',\n",
       "  'rs141432255',\n",
       "  'rs574353697',\n",
       "  'rs202085145',\n",
       "  'rs368327991',\n",
       "  'rs370951378',\n",
       "  'rs748900391',\n",
       "  'rs747461754',\n",
       "  'rs770496917',\n",
       "  'rs770792767',\n",
       "  'rs794729211',\n",
       "  'rs773937413',\n",
       "  'rs765738385',\n",
       "  'rs766949012',\n",
       "  'rs780603746',\n",
       "  'rs781081194',\n",
       "  'rs781351306',\n",
       "  'rs759992645',\n",
       "  'rs760901724',\n",
       "  'rs778431071',\n",
       "  'rs765012855',\n",
       "  'rs776356158',\n",
       "  'rs776690106',\n",
       "  'rs1171191979',\n",
       "  'rs1175117475',\n",
       "  'rs1273768268',\n",
       "  'rs1372045486',\n",
       "  'rs1370270518',\n",
       "  'rs1435587153',\n",
       "  'rs1451449404',\n",
       "  'rs1460210127',\n",
       "  'rs1565798201',\n",
       "  'rs1565798380',\n",
       "  'rs1565799921',\n",
       "  'rs1555198521',\n",
       "  'rs1592831898',\n",
       "  'rs1890630965',\n",
       "  'rs1890647333',\n",
       "  'rs1890663087',\n",
       "  'rs1890663871',\n",
       "  'rs1890688726',\n",
       "  'rs2136509616',\n",
       "  'rs2136509843',\n",
       "  'rs2136512310',\n",
       "  'rs2136513040',\n",
       "  'rs2136513148',\n",
       "  'rs2136513466'],\n",
       " 'ENSG00000140090': ['rs141131742',\n",
       "  'rs587777535',\n",
       "  'rs587777536',\n",
       "  'rs587777537',\n",
       "  'rs1595312054'],\n",
       " 'ENSG00000141252': ['rs561656938',\n",
       "  'rs370828258',\n",
       "  'rs587777465',\n",
       "  'rs587777466',\n",
       "  'rs200594402',\n",
       "  'rs768997239',\n",
       "  'rs776733428',\n",
       "  'rs1335346154',\n",
       "  'rs1447478732',\n",
       "  'rs1472685858'],\n",
       " 'ENSG00000145335': ['rs104893875',\n",
       "  'rs104893877',\n",
       "  'rs104893878',\n",
       "  'rs201106962',\n",
       "  'rs431905511',\n",
       "  'rs1739238968',\n",
       "  'rs2110525413'],\n",
       " 'ENSG00000165916': ['rs1177898071', 'rs1363348500'],\n",
       " 'ENSG00000186153': ['rs75559202',\n",
       "  'rs79771882',\n",
       "  'rs193027041',\n",
       "  'rs186745328',\n",
       "  'rs144234059',\n",
       "  'rs144412859',\n",
       "  'rs117209694',\n",
       "  'rs119487098',\n",
       "  'rs139253468',\n",
       "  'rs587777127',\n",
       "  'rs587777128',\n",
       "  'rs587777248',\n",
       "  'rs730880290',\n",
       "  'rs730880291',\n",
       "  'rs730880292',\n",
       "  'rs730882215',\n",
       "  'rs375757102',\n",
       "  'rs368928190',\n",
       "  'rs199511589',\n",
       "  'rs199628364',\n",
       "  'rs200812468',\n",
       "  'rs200839945',\n",
       "  'rs201008667',\n",
       "  'rs751181600',\n",
       "  'rs752354290',\n",
       "  'rs761879076',\n",
       "  'rs775696083',\n",
       "  'rs765339835',\n",
       "  'rs765544833',\n",
       "  'rs758588684',\n",
       "  'rs759766243',\n",
       "  'rs767732033',\n",
       "  'rs770023814',\n",
       "  'rs770155582',\n",
       "  'rs867163041',\n",
       "  'rs778218106',\n",
       "  'rs780345312',\n",
       "  'rs878855021',\n",
       "  'rs886039653',\n",
       "  'rs373306276',\n",
       "  'rs756762196',\n",
       "  'rs990150249',\n",
       "  'rs1039151413',\n",
       "  'rs1164465811',\n",
       "  'rs1064795117',\n",
       "  'rs1131691285',\n",
       "  'rs1057517846',\n",
       "  'rs1057518795',\n",
       "  'rs1060502727',\n",
       "  'rs1232899835',\n",
       "  'rs1284883505',\n",
       "  'rs1295198168',\n",
       "  'rs1300924648',\n",
       "  'rs1394607357',\n",
       "  'rs1377640182',\n",
       "  'rs1484856529',\n",
       "  'rs1555532979',\n",
       "  'rs1555535072',\n",
       "  'rs1567542020',\n",
       "  'rs1567567249',\n",
       "  'rs1567625065',\n",
       "  'rs1597207802',\n",
       "  'rs1597207871',\n",
       "  'rs1597216056',\n",
       "  'rs2032287989',\n",
       "  'rs2051753978',\n",
       "  'rs2051756247',\n",
       "  'rs2083042603',\n",
       "  'rs2151934194',\n",
       "  'rs2151674791',\n",
       "  'rs2151386374']}"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a1ac318b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "110e80e3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "bd39b773",
   "metadata": {},
   "source": [
    "## 2. Filtering chr tsv files and obtaining iid-snps matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6317c4f5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# TSV files were generated straight from the chr_dosage_gz files\n",
    "# Script checks which rows in the tsv files are SNPs from the SNP dictionary\n",
    "# At last, it creates a matrix which rows are rosmap individuals and cols are snps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "bdfe1754",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the gene-variant dictionary from the pickle file\n",
    "# old dictionary.\n",
    "\n",
    "import pickle\n",
    "\n",
    "dict_path = './ad_shap_gwas/snps_list/gene_variants_dict_AD_filt_phenotype.pkl'\n",
    "with open(dict_path, 'rb') as f:\n",
    "    gene_variants_dict = pickle.load(f)\n",
    "\n",
    "# Collect all SNP IDs from the dictionary into a set for fast lookup\n",
    "snp_set = set()\n",
    "for snps in gene_variants_dict.values():\n",
    "    snp_set.update(snps)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "956aa472",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "24078"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(snp_set)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "8e56c534",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('./ad_shap_gwas//dosage_imputed_vcf_files/dosage_gz_files/AMP-AD_ROSMAP_Rush-Broad_AffymetrixGenechip6_Imputed_chr22.tsv', sep=' ', header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "0a5a8cad",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>1702</th>\n",
       "      <th>1703</th>\n",
       "      <th>1704</th>\n",
       "      <th>1705</th>\n",
       "      <th>1706</th>\n",
       "      <th>1707</th>\n",
       "      <th>1708</th>\n",
       "      <th>1709</th>\n",
       "      <th>1710</th>\n",
       "      <th>1711</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs62224609</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1.90811</td>\n",
       "      <td>1.90858</td>\n",
       "      <td>1.90811</td>\n",
       "      <td>1.90904</td>\n",
       "      <td>1.90811</td>\n",
       "      <td>1.90811</td>\n",
       "      <td>1.90904</td>\n",
       "      <td>...</td>\n",
       "      <td>1.90941</td>\n",
       "      <td>1.90941</td>\n",
       "      <td>1.90958</td>\n",
       "      <td>1.90941</td>\n",
       "      <td>1.90959</td>\n",
       "      <td>1.90941</td>\n",
       "      <td>1.90958</td>\n",
       "      <td>1.90959</td>\n",
       "      <td>1.90958</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs11089130</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1.65161</td>\n",
       "      <td>1.65336</td>\n",
       "      <td>1.65161</td>\n",
       "      <td>1.65512</td>\n",
       "      <td>1.65161</td>\n",
       "      <td>1.65161</td>\n",
       "      <td>1.65511</td>\n",
       "      <td>...</td>\n",
       "      <td>1.66232</td>\n",
       "      <td>1.66232</td>\n",
       "      <td>1.66300</td>\n",
       "      <td>1.66232</td>\n",
       "      <td>1.66299</td>\n",
       "      <td>1.66232</td>\n",
       "      <td>1.66300</td>\n",
       "      <td>1.66299</td>\n",
       "      <td>1.66300</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs11089134</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.64777</td>\n",
       "      <td>1.64953</td>\n",
       "      <td>1.64777</td>\n",
       "      <td>1.65132</td>\n",
       "      <td>1.64777</td>\n",
       "      <td>1.64777</td>\n",
       "      <td>1.65131</td>\n",
       "      <td>...</td>\n",
       "      <td>1.65757</td>\n",
       "      <td>1.65757</td>\n",
       "      <td>1.65825</td>\n",
       "      <td>1.65757</td>\n",
       "      <td>1.65823</td>\n",
       "      <td>1.65757</td>\n",
       "      <td>1.65826</td>\n",
       "      <td>1.65823</td>\n",
       "      <td>1.65825</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs2843213</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1.94153</td>\n",
       "      <td>1.94165</td>\n",
       "      <td>1.94153</td>\n",
       "      <td>1.94179</td>\n",
       "      <td>1.94153</td>\n",
       "      <td>1.94153</td>\n",
       "      <td>1.94179</td>\n",
       "      <td>...</td>\n",
       "      <td>1.94326</td>\n",
       "      <td>1.94326</td>\n",
       "      <td>1.94330</td>\n",
       "      <td>1.94326</td>\n",
       "      <td>1.94330</td>\n",
       "      <td>1.94326</td>\n",
       "      <td>1.94330</td>\n",
       "      <td>1.94330</td>\n",
       "      <td>1.94330</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs2096600</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>1.31489</td>\n",
       "      <td>1.31839</td>\n",
       "      <td>1.31489</td>\n",
       "      <td>1.32192</td>\n",
       "      <td>1.31489</td>\n",
       "      <td>1.31489</td>\n",
       "      <td>1.32191</td>\n",
       "      <td>...</td>\n",
       "      <td>1.32288</td>\n",
       "      <td>1.32288</td>\n",
       "      <td>1.32421</td>\n",
       "      <td>1.32288</td>\n",
       "      <td>1.32419</td>\n",
       "      <td>1.32288</td>\n",
       "      <td>1.32422</td>\n",
       "      <td>1.32419</td>\n",
       "      <td>1.32421</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 1712 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         0    1    2        3        4        5        6        7        8     \\\n",
       "0  rs62224609    T    C  1.90811  1.90858  1.90811  1.90904  1.90811  1.90811   \n",
       "1  rs11089130    G    C  1.65161  1.65336  1.65161  1.65512  1.65161  1.65161   \n",
       "2  rs11089134    A    G  1.64777  1.64953  1.64777  1.65132  1.64777  1.64777   \n",
       "3   rs2843213    C    T  1.94153  1.94165  1.94153  1.94179  1.94153  1.94153   \n",
       "4   rs2096600    A    C  1.31489  1.31839  1.31489  1.32192  1.31489  1.31489   \n",
       "\n",
       "      9     ...     1702     1703     1704     1705     1706     1707  \\\n",
       "0  1.90904  ...  1.90941  1.90941  1.90958  1.90941  1.90959  1.90941   \n",
       "1  1.65511  ...  1.66232  1.66232  1.66300  1.66232  1.66299  1.66232   \n",
       "2  1.65131  ...  1.65757  1.65757  1.65825  1.65757  1.65823  1.65757   \n",
       "3  1.94179  ...  1.94326  1.94326  1.94330  1.94326  1.94330  1.94326   \n",
       "4  1.32191  ...  1.32288  1.32288  1.32421  1.32288  1.32419  1.32288   \n",
       "\n",
       "      1708     1709     1710  1711  \n",
       "0  1.90958  1.90959  1.90958   NaN  \n",
       "1  1.66300  1.66299  1.66300   NaN  \n",
       "2  1.65826  1.65823  1.65825   NaN  \n",
       "3  1.94330  1.94330  1.94330   NaN  \n",
       "4  1.32422  1.32419  1.32421   NaN  \n",
       "\n",
       "[5 rows x 1712 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aa728f2d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fb231f63",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import pickle\n",
    "\n",
    "# Load the gene-variant dictionary from the pickle file\n",
    "dict_path = '/home/vmottaqi/rnaseq_synapse/ad_shap_gwas/snps_list/gene_variants_dict_AD_filt_clinical.pkl'\n",
    "with open(dict_path, 'rb') as f:\n",
    "    gene_variants_dict = pickle.load(f)\n",
    "\n",
    "# Collect all SNP IDs from the dictionary into a set for fast lookup\n",
    "snp_set = set()\n",
    "for snps in gene_variants_dict.values():\n",
    "    snp_set.update(snps)\n",
    "\n",
    "# Directory containing the TSV files\n",
    "directory = '/home/vmottaqi/rnaseq_synapse/ad_shap_gwas/dosage_imputed_vcf_files/dosage_gz_files/'\n",
    "directory1 = \"/home/vmottaqi/rnaseq_synapse/ad_shap_gwas/dosage_imputed_vcf_files/chr_files_tsv/\"\n",
    "\n",
    "# Loop through all TSV files in the directory\n",
    "for filename in os.listdir(directory):\n",
    "    if filename.endswith('.tsv'):\n",
    "        filepath = os.path.join(directory, filename)\n",
    "        df = pd.read_csv(filepath, sep=' ', header=None)\n",
    "\n",
    "        # Filter the dataframe to keep only rows where the first column is in snp_set\n",
    "        filtered_df = df[df[0].isin(snp_set)]\n",
    "\n",
    "        # Count the rows in the filtered DataFrame\n",
    "        row_count = filtered_df.shape[0]\n",
    "\n",
    "        # Create new filename based on the last part of the original filename\n",
    "        new_filename = filename.split('_')[-1]\n",
    "        new_filepath = os.path.join(directory1, new_filename)\n",
    "\n",
    "        # Save the filtered dataframe to a new TSV file\n",
    "        filtered_df.to_csv(new_filepath, sep='\\t', index=False, header=False)\n",
    "\n",
    "        # Print information about the processing and row count\n",
    "        print(f\"Processed {filename}: {row_count} rows saved to {new_filename}\")\n",
    "\n",
    "print(\"All files have been processed.\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "967860dd",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import pickle\n",
    "from joblib import Parallel, delayed\n",
    "\n",
    "def process_file(filename, directory, directory1, snp_set):\n",
    "    filepath = os.path.join(directory, filename)\n",
    "    df = pd.read_csv(filepath, sep=' ', header=None)\n",
    "\n",
    "    # Filter the dataframe to keep only rows where the first column is in snp_set\n",
    "    filtered_df = df[df[0].isin(snp_set)]\n",
    "\n",
    "    # Count the rows in the filtered DataFrame\n",
    "    row_count = filtered_df.shape[0]\n",
    "\n",
    "    # Create new filename based on the last part of the original filename\n",
    "    new_filename = filename.split('_')[-1]\n",
    "    new_filepath = os.path.join(directory1, new_filename)\n",
    "\n",
    "    # Save the filtered dataframe to a new TSV file\n",
    "    filtered_df.to_csv(new_filepath, sep='\\t', index=False, header=False)\n",
    "\n",
    "    # Print information about the processing and row count\n",
    "    return f\"Processed {filename}: {row_count} rows saved to {new_filename}\"\n",
    "\n",
    "# Load the gene-variant dictionary from the pickle file\n",
    "dict_path = '/home/vmottaqi/rnaseq_synapse/ad_shap_gwas/snps_list/gene_variants_dict_AD_filt_clinical.pkl'\n",
    "with open(dict_path, 'rb') as f:\n",
    "    gene_variants_dict = pickle.load(f)\n",
    "\n",
    "# Collect all SNP IDs from the dictionary into a set for fast lookup\n",
    "snp_set = set()\n",
    "for snps in gene_variants_dict.values():\n",
    "    snp_set.update(snps)\n",
    "\n",
    "# Directory containing the TSV files\n",
    "directory = '/home/vmottaqi/rnaseq_synapse/ad_shap_gwas/dosage_imputed_vcf_files/dosage_gz_files/'\n",
    "directory1 = \"/home/vmottaqi/rnaseq_synapse/ad_shap_gwas/dosage_imputed_vcf_files/chr_files_tsv/\"\n",
    "\n",
    "# Get all TSV files\n",
    "tsv_files = [f for f in os.listdir(directory) if f.endswith('.tsv')]\n",
    "\n",
    "# Execute in parallel using joblib\n",
    "results = Parallel(n_jobs=8)(delayed(process_file)(filename, directory, directory1, snp_set) for filename in tsv_files)\n",
    "\n",
    "# Print results\n",
    "for result in results:\n",
    "    print(result)\n",
    "\n",
    "print(\"All files have been processed.\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "93d09d82",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5b0266cd",
   "metadata": {},
   "outputs": [],
   "source": [
    "# UNIX command to merge the tsv files into one tsv file (row-wise)\n",
    "cat /path/to/directory/*.tsv > merged_file.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "96d15283",
   "metadata": {},
   "outputs": [],
   "source": [
    "# just a check to see how many of the rows in the chr_concatinated can be mapped to my new dictionary (clinic)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "43bca805",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "13297c11",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e98bfbe1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add header as the list of individuals to the tsv file\n",
    "# Filter and keep only ROSMAP individuals\n",
    "# Transpose the dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5d658105",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Step by step"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a674ec4f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Load the TSV file without headers\n",
    "tsv_file_path = '/home/vmottaqi/rnaseq_synapse/ad_shap_gwas/dosage_imputed_vcf_files/chr_concatinated.tsv'\n",
    "df = pd.read_csv(tsv_file_path, sep='\\s+', header=None, engine='python')\n",
    "\n",
    "# Load the individual identifiers\n",
    "individuals_file_path = '/home/vmottaqi/rnaseq_synapse/ad_shap_gwas/dosage_imputed_vcf_files/individuals.csv'\n",
    "with open(individuals_file_path, 'r') as file:\n",
    "    # Assuming each line contains one individual identifier\n",
    "    headers = [line.strip() for line in file.readlines()]\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0edcfcff",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>1701</th>\n",
       "      <th>1702</th>\n",
       "      <th>1703</th>\n",
       "      <th>1704</th>\n",
       "      <th>1705</th>\n",
       "      <th>1706</th>\n",
       "      <th>1707</th>\n",
       "      <th>1708</th>\n",
       "      <th>1709</th>\n",
       "      <th>1710</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs72777665</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.28960</td>\n",
       "      <td>1.28864</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.28953</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.28922</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs35345466</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>0.93633</td>\n",
       "      <td>0.26569</td>\n",
       "      <td>0.03411</td>\n",
       "      <td>0.93413</td>\n",
       "      <td>0.95241</td>\n",
       "      <td>0.96472</td>\n",
       "      <td>0.94774</td>\n",
       "      <td>...</td>\n",
       "      <td>0.93148</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.92957</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.14522</td>\n",
       "      <td>1.84339</td>\n",
       "      <td>0.91914</td>\n",
       "      <td>1.15320</td>\n",
       "      <td>0.16229</td>\n",
       "      <td>0.16842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs11257183</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs16914105</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs7917618</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.99896</td>\n",
       "      <td>1.99931</td>\n",
       "      <td>1.99999</td>\n",
       "      <td>1.99999</td>\n",
       "      <td>1.99907</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.99911</td>\n",
       "      <td>1.99965</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.99992</td>\n",
       "      <td>1.99649</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.99894</td>\n",
       "      <td>1.99820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2960</th>\n",
       "      <td>rs36137267</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1.02021</td>\n",
       "      <td>1.01901</td>\n",
       "      <td>0.98175</td>\n",
       "      <td>1.01908</td>\n",
       "      <td>1.99010</td>\n",
       "      <td>0.97431</td>\n",
       "      <td>1.97976</td>\n",
       "      <td>...</td>\n",
       "      <td>1.94194</td>\n",
       "      <td>0.03687</td>\n",
       "      <td>0.07033</td>\n",
       "      <td>0.09149</td>\n",
       "      <td>1.05846</td>\n",
       "      <td>1.01997</td>\n",
       "      <td>1.06632</td>\n",
       "      <td>1.85975</td>\n",
       "      <td>1.02561</td>\n",
       "      <td>1.97771</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2961</th>\n",
       "      <td>rs6606291</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1.89330</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>1.89330</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>...</td>\n",
       "      <td>1.88489</td>\n",
       "      <td>1.89351</td>\n",
       "      <td>1.88489</td>\n",
       "      <td>1.89351</td>\n",
       "      <td>1.88489</td>\n",
       "      <td>1.88489</td>\n",
       "      <td>1.88484</td>\n",
       "      <td>1.89351</td>\n",
       "      <td>1.89351</td>\n",
       "      <td>1.88489</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2962</th>\n",
       "      <td>rs12337435</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1.00350</td>\n",
       "      <td>1.00536</td>\n",
       "      <td>1.99921</td>\n",
       "      <td>0.02163</td>\n",
       "      <td>1.99959</td>\n",
       "      <td>0.02076</td>\n",
       "      <td>1.00962</td>\n",
       "      <td>...</td>\n",
       "      <td>1.01501</td>\n",
       "      <td>1.99907</td>\n",
       "      <td>1.01535</td>\n",
       "      <td>1.99949</td>\n",
       "      <td>1.01626</td>\n",
       "      <td>1.00376</td>\n",
       "      <td>1.99873</td>\n",
       "      <td>1.00267</td>\n",
       "      <td>1.01543</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2963</th>\n",
       "      <td>rs12343184</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1.00089</td>\n",
       "      <td>1.00113</td>\n",
       "      <td>1.99937</td>\n",
       "      <td>0.01343</td>\n",
       "      <td>1.99959</td>\n",
       "      <td>0.01337</td>\n",
       "      <td>1.00651</td>\n",
       "      <td>...</td>\n",
       "      <td>1.01438</td>\n",
       "      <td>1.99923</td>\n",
       "      <td>1.01441</td>\n",
       "      <td>1.99969</td>\n",
       "      <td>1.01460</td>\n",
       "      <td>1.00085</td>\n",
       "      <td>1.99906</td>\n",
       "      <td>1.00123</td>\n",
       "      <td>1.01452</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2964</th>\n",
       "      <td>rs28886421</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.98999</td>\n",
       "      <td>1.99000</td>\n",
       "      <td>1.97996</td>\n",
       "      <td>1.97997</td>\n",
       "      <td>1.98999</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.99045</td>\n",
       "      <td>1.98954</td>\n",
       "      <td>1.97913</td>\n",
       "      <td>1.98168</td>\n",
       "      <td>1.98959</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.98954</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2965 rows × 1711 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            0    1    2        3        4        5        6        7     \\\n",
       "0     rs72777665    T    A  2.00000  2.00000  2.00000  2.00000  2.00000   \n",
       "1     rs35345466    A    T  0.93633  0.26569  0.03411  0.93413  0.95241   \n",
       "2     rs11257183    T    C  1.00000  2.00000  2.00000  1.00000  2.00000   \n",
       "3     rs16914105    C    T  1.00000  2.00000  2.00000  2.00000  2.00000   \n",
       "4      rs7917618    G    A  2.00000  1.99896  1.99931  1.99999  1.99999   \n",
       "...          ...  ...  ...      ...      ...      ...      ...      ...   \n",
       "2960  rs36137267    T    C  1.02021  1.01901  0.98175  1.01908  1.99010   \n",
       "2961   rs6606291    C    T  1.89330  1.88121  1.88121  1.88121  1.88121   \n",
       "2962  rs12337435    C    A  1.00350  1.00536  1.99921  0.02163  1.99959   \n",
       "2963  rs12343184    T    C  1.00089  1.00113  1.99937  0.01343  1.99959   \n",
       "2964  rs28886421    A    G  2.00000  1.98999  1.99000  1.97996  1.97997   \n",
       "\n",
       "         8        9     ...     1701     1702     1703     1704     1705  \\\n",
       "0     2.00000  2.00000  ...  1.28960  1.28864  2.00000  1.28953  2.00000   \n",
       "1     0.96472  0.94774  ...  0.93148  0.00000  0.92957  0.00000  0.14522   \n",
       "2     2.00000  2.00000  ...  2.00000  2.00000  1.00000  1.00000  2.00000   \n",
       "3     2.00000  2.00000  ...  2.00000  2.00000  2.00000  2.00000  2.00000   \n",
       "4     1.99907  2.00000  ...  1.99911  1.99965  2.00000  2.00000  2.00000   \n",
       "...       ...      ...  ...      ...      ...      ...      ...      ...   \n",
       "2960  0.97431  1.97976  ...  1.94194  0.03687  0.07033  0.09149  1.05846   \n",
       "2961  1.89330  1.88121  ...  1.88489  1.89351  1.88489  1.89351  1.88489   \n",
       "2962  0.02076  1.00962  ...  1.01501  1.99907  1.01535  1.99949  1.01626   \n",
       "2963  0.01337  1.00651  ...  1.01438  1.99923  1.01441  1.99969  1.01460   \n",
       "2964  1.98999  2.00000  ...  1.99045  1.98954  1.97913  1.98168  1.98959   \n",
       "\n",
       "         1706     1707     1708     1709     1710  \n",
       "0     2.00000  2.00000  2.00000  1.28922  2.00000  \n",
       "1     1.84339  0.91914  1.15320  0.16229  0.16842  \n",
       "2     2.00000  1.00000  2.00000  2.00000  1.00000  \n",
       "3     1.00000  2.00000  1.00000  2.00000  2.00000  \n",
       "4     1.99992  1.99649  2.00000  1.99894  1.99820  \n",
       "...       ...      ...      ...      ...      ...  \n",
       "2960  1.01997  1.06632  1.85975  1.02561  1.97771  \n",
       "2961  1.88489  1.88484  1.89351  1.89351  1.88489  \n",
       "2962  1.00376  1.99873  1.00267  1.01543  2.00000  \n",
       "2963  1.00085  1.99906  1.00123  1.01452  2.00000  \n",
       "2964  2.00000  1.98954  2.00000  2.00000  2.00000  \n",
       "\n",
       "[2965 rows x 1711 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "10b39e9f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "307bbc1e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Assign the headers to the dataframe\n",
    "df.columns = ['SNP_ID', \"REF\", \"ALT\"] + headers  # Prepend 'SNP_ID' for the first column\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9d5fa360",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP_ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>ROS10442701</th>\n",
       "      <th>ROS11444465</th>\n",
       "      <th>ROS21112011</th>\n",
       "      <th>ROS20297791</th>\n",
       "      <th>ROS11395417</th>\n",
       "      <th>ROS21174303</th>\n",
       "      <th>ROS10669174</th>\n",
       "      <th>...</th>\n",
       "      <th>MAP50406183</th>\n",
       "      <th>MAP50401426</th>\n",
       "      <th>ROS10263208</th>\n",
       "      <th>MAP50109053</th>\n",
       "      <th>MAP50104396</th>\n",
       "      <th>MAP33006377</th>\n",
       "      <th>MAP50402693</th>\n",
       "      <th>ROS11349821</th>\n",
       "      <th>ROS11099004</th>\n",
       "      <th>MAP29821047</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs72777665</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.28960</td>\n",
       "      <td>1.28864</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.28953</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.28922</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs35345466</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>0.93633</td>\n",
       "      <td>0.26569</td>\n",
       "      <td>0.03411</td>\n",
       "      <td>0.93413</td>\n",
       "      <td>0.95241</td>\n",
       "      <td>0.96472</td>\n",
       "      <td>0.94774</td>\n",
       "      <td>...</td>\n",
       "      <td>0.93148</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.92957</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.14522</td>\n",
       "      <td>1.84339</td>\n",
       "      <td>0.91914</td>\n",
       "      <td>1.15320</td>\n",
       "      <td>0.16229</td>\n",
       "      <td>0.16842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs11257183</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs16914105</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs7917618</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.99896</td>\n",
       "      <td>1.99931</td>\n",
       "      <td>1.99999</td>\n",
       "      <td>1.99999</td>\n",
       "      <td>1.99907</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.99911</td>\n",
       "      <td>1.99965</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.99992</td>\n",
       "      <td>1.99649</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.99894</td>\n",
       "      <td>1.99820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2960</th>\n",
       "      <td>rs36137267</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1.02021</td>\n",
       "      <td>1.01901</td>\n",
       "      <td>0.98175</td>\n",
       "      <td>1.01908</td>\n",
       "      <td>1.99010</td>\n",
       "      <td>0.97431</td>\n",
       "      <td>1.97976</td>\n",
       "      <td>...</td>\n",
       "      <td>1.94194</td>\n",
       "      <td>0.03687</td>\n",
       "      <td>0.07033</td>\n",
       "      <td>0.09149</td>\n",
       "      <td>1.05846</td>\n",
       "      <td>1.01997</td>\n",
       "      <td>1.06632</td>\n",
       "      <td>1.85975</td>\n",
       "      <td>1.02561</td>\n",
       "      <td>1.97771</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2961</th>\n",
       "      <td>rs6606291</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1.89330</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>1.89330</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>...</td>\n",
       "      <td>1.88489</td>\n",
       "      <td>1.89351</td>\n",
       "      <td>1.88489</td>\n",
       "      <td>1.89351</td>\n",
       "      <td>1.88489</td>\n",
       "      <td>1.88489</td>\n",
       "      <td>1.88484</td>\n",
       "      <td>1.89351</td>\n",
       "      <td>1.89351</td>\n",
       "      <td>1.88489</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2962</th>\n",
       "      <td>rs12337435</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1.00350</td>\n",
       "      <td>1.00536</td>\n",
       "      <td>1.99921</td>\n",
       "      <td>0.02163</td>\n",
       "      <td>1.99959</td>\n",
       "      <td>0.02076</td>\n",
       "      <td>1.00962</td>\n",
       "      <td>...</td>\n",
       "      <td>1.01501</td>\n",
       "      <td>1.99907</td>\n",
       "      <td>1.01535</td>\n",
       "      <td>1.99949</td>\n",
       "      <td>1.01626</td>\n",
       "      <td>1.00376</td>\n",
       "      <td>1.99873</td>\n",
       "      <td>1.00267</td>\n",
       "      <td>1.01543</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2963</th>\n",
       "      <td>rs12343184</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1.00089</td>\n",
       "      <td>1.00113</td>\n",
       "      <td>1.99937</td>\n",
       "      <td>0.01343</td>\n",
       "      <td>1.99959</td>\n",
       "      <td>0.01337</td>\n",
       "      <td>1.00651</td>\n",
       "      <td>...</td>\n",
       "      <td>1.01438</td>\n",
       "      <td>1.99923</td>\n",
       "      <td>1.01441</td>\n",
       "      <td>1.99969</td>\n",
       "      <td>1.01460</td>\n",
       "      <td>1.00085</td>\n",
       "      <td>1.99906</td>\n",
       "      <td>1.00123</td>\n",
       "      <td>1.01452</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2964</th>\n",
       "      <td>rs28886421</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.98999</td>\n",
       "      <td>1.99000</td>\n",
       "      <td>1.97996</td>\n",
       "      <td>1.97997</td>\n",
       "      <td>1.98999</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.99045</td>\n",
       "      <td>1.98954</td>\n",
       "      <td>1.97913</td>\n",
       "      <td>1.98168</td>\n",
       "      <td>1.98959</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.98954</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2965 rows × 1711 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          SNP_ID REF ALT  ROS10442701  ROS11444465  ROS21112011  ROS20297791  \\\n",
       "0     rs72777665   T   A      2.00000      2.00000      2.00000      2.00000   \n",
       "1     rs35345466   A   T      0.93633      0.26569      0.03411      0.93413   \n",
       "2     rs11257183   T   C      1.00000      2.00000      2.00000      1.00000   \n",
       "3     rs16914105   C   T      1.00000      2.00000      2.00000      2.00000   \n",
       "4      rs7917618   G   A      2.00000      1.99896      1.99931      1.99999   \n",
       "...          ...  ..  ..          ...          ...          ...          ...   \n",
       "2960  rs36137267   T   C      1.02021      1.01901      0.98175      1.01908   \n",
       "2961   rs6606291   C   T      1.89330      1.88121      1.88121      1.88121   \n",
       "2962  rs12337435   C   A      1.00350      1.00536      1.99921      0.02163   \n",
       "2963  rs12343184   T   C      1.00089      1.00113      1.99937      0.01343   \n",
       "2964  rs28886421   A   G      2.00000      1.98999      1.99000      1.97996   \n",
       "\n",
       "      ROS11395417  ROS21174303  ROS10669174  ...  MAP50406183  MAP50401426  \\\n",
       "0         2.00000      2.00000      2.00000  ...      1.28960      1.28864   \n",
       "1         0.95241      0.96472      0.94774  ...      0.93148      0.00000   \n",
       "2         2.00000      2.00000      2.00000  ...      2.00000      2.00000   \n",
       "3         2.00000      2.00000      2.00000  ...      2.00000      2.00000   \n",
       "4         1.99999      1.99907      2.00000  ...      1.99911      1.99965   \n",
       "...           ...          ...          ...  ...          ...          ...   \n",
       "2960      1.99010      0.97431      1.97976  ...      1.94194      0.03687   \n",
       "2961      1.88121      1.89330      1.88121  ...      1.88489      1.89351   \n",
       "2962      1.99959      0.02076      1.00962  ...      1.01501      1.99907   \n",
       "2963      1.99959      0.01337      1.00651  ...      1.01438      1.99923   \n",
       "2964      1.97997      1.98999      2.00000  ...      1.99045      1.98954   \n",
       "\n",
       "      ROS10263208  MAP50109053  MAP50104396  MAP33006377  MAP50402693  \\\n",
       "0         2.00000      1.28953      2.00000      2.00000      2.00000   \n",
       "1         0.92957      0.00000      0.14522      1.84339      0.91914   \n",
       "2         1.00000      1.00000      2.00000      2.00000      1.00000   \n",
       "3         2.00000      2.00000      2.00000      1.00000      2.00000   \n",
       "4         2.00000      2.00000      2.00000      1.99992      1.99649   \n",
       "...           ...          ...          ...          ...          ...   \n",
       "2960      0.07033      0.09149      1.05846      1.01997      1.06632   \n",
       "2961      1.88489      1.89351      1.88489      1.88489      1.88484   \n",
       "2962      1.01535      1.99949      1.01626      1.00376      1.99873   \n",
       "2963      1.01441      1.99969      1.01460      1.00085      1.99906   \n",
       "2964      1.97913      1.98168      1.98959      2.00000      1.98954   \n",
       "\n",
       "      ROS11349821  ROS11099004  MAP29821047  \n",
       "0         2.00000      1.28922      2.00000  \n",
       "1         1.15320      0.16229      0.16842  \n",
       "2         2.00000      2.00000      1.00000  \n",
       "3         1.00000      2.00000      2.00000  \n",
       "4         2.00000      1.99894      1.99820  \n",
       "...           ...          ...          ...  \n",
       "2960      1.85975      1.02561      1.97771  \n",
       "2961      1.89351      1.89351      1.88489  \n",
       "2962      1.00267      1.01543      2.00000  \n",
       "2963      1.00123      1.01452      2.00000  \n",
       "2964      2.00000      2.00000      2.00000  \n",
       "\n",
       "[2965 rows x 1711 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a67d7da8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Function to remove the prefix \"ROS\" or \"MAP\" from column names\n",
    "def remove_prefix(col_name):\n",
    "    if col_name.startswith(\"ROS\"):\n",
    "        return col_name[3:]  # Remove the first 3 characters\n",
    "    elif col_name.startswith(\"MAP\"):\n",
    "        return col_name[3:]  # Remove the first 3 characters\n",
    "    else:\n",
    "        return col_name\n",
    "\n",
    "# Apply the function to each column name\n",
    "df.columns = [remove_prefix(col) for col in df.columns]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "be6c7c0d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP_ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>10442701</th>\n",
       "      <th>11444465</th>\n",
       "      <th>21112011</th>\n",
       "      <th>20297791</th>\n",
       "      <th>11395417</th>\n",
       "      <th>21174303</th>\n",
       "      <th>10669174</th>\n",
       "      <th>...</th>\n",
       "      <th>50406183</th>\n",
       "      <th>50401426</th>\n",
       "      <th>10263208</th>\n",
       "      <th>50109053</th>\n",
       "      <th>50104396</th>\n",
       "      <th>33006377</th>\n",
       "      <th>50402693</th>\n",
       "      <th>11349821</th>\n",
       "      <th>11099004</th>\n",
       "      <th>29821047</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs72777665</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.28960</td>\n",
       "      <td>1.28864</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.28953</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.28922</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs35345466</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>0.93633</td>\n",
       "      <td>0.26569</td>\n",
       "      <td>0.03411</td>\n",
       "      <td>0.93413</td>\n",
       "      <td>0.95241</td>\n",
       "      <td>0.96472</td>\n",
       "      <td>0.94774</td>\n",
       "      <td>...</td>\n",
       "      <td>0.93148</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.92957</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.14522</td>\n",
       "      <td>1.84339</td>\n",
       "      <td>0.91914</td>\n",
       "      <td>1.15320</td>\n",
       "      <td>0.16229</td>\n",
       "      <td>0.16842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs11257183</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs16914105</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs7917618</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.99896</td>\n",
       "      <td>1.99931</td>\n",
       "      <td>1.99999</td>\n",
       "      <td>1.99999</td>\n",
       "      <td>1.99907</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.99911</td>\n",
       "      <td>1.99965</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.99992</td>\n",
       "      <td>1.99649</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.99894</td>\n",
       "      <td>1.99820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2960</th>\n",
       "      <td>rs36137267</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1.02021</td>\n",
       "      <td>1.01901</td>\n",
       "      <td>0.98175</td>\n",
       "      <td>1.01908</td>\n",
       "      <td>1.99010</td>\n",
       "      <td>0.97431</td>\n",
       "      <td>1.97976</td>\n",
       "      <td>...</td>\n",
       "      <td>1.94194</td>\n",
       "      <td>0.03687</td>\n",
       "      <td>0.07033</td>\n",
       "      <td>0.09149</td>\n",
       "      <td>1.05846</td>\n",
       "      <td>1.01997</td>\n",
       "      <td>1.06632</td>\n",
       "      <td>1.85975</td>\n",
       "      <td>1.02561</td>\n",
       "      <td>1.97771</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2961</th>\n",
       "      <td>rs6606291</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1.89330</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>1.89330</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>...</td>\n",
       "      <td>1.88489</td>\n",
       "      <td>1.89351</td>\n",
       "      <td>1.88489</td>\n",
       "      <td>1.89351</td>\n",
       "      <td>1.88489</td>\n",
       "      <td>1.88489</td>\n",
       "      <td>1.88484</td>\n",
       "      <td>1.89351</td>\n",
       "      <td>1.89351</td>\n",
       "      <td>1.88489</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2962</th>\n",
       "      <td>rs12337435</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1.00350</td>\n",
       "      <td>1.00536</td>\n",
       "      <td>1.99921</td>\n",
       "      <td>0.02163</td>\n",
       "      <td>1.99959</td>\n",
       "      <td>0.02076</td>\n",
       "      <td>1.00962</td>\n",
       "      <td>...</td>\n",
       "      <td>1.01501</td>\n",
       "      <td>1.99907</td>\n",
       "      <td>1.01535</td>\n",
       "      <td>1.99949</td>\n",
       "      <td>1.01626</td>\n",
       "      <td>1.00376</td>\n",
       "      <td>1.99873</td>\n",
       "      <td>1.00267</td>\n",
       "      <td>1.01543</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2963</th>\n",
       "      <td>rs12343184</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1.00089</td>\n",
       "      <td>1.00113</td>\n",
       "      <td>1.99937</td>\n",
       "      <td>0.01343</td>\n",
       "      <td>1.99959</td>\n",
       "      <td>0.01337</td>\n",
       "      <td>1.00651</td>\n",
       "      <td>...</td>\n",
       "      <td>1.01438</td>\n",
       "      <td>1.99923</td>\n",
       "      <td>1.01441</td>\n",
       "      <td>1.99969</td>\n",
       "      <td>1.01460</td>\n",
       "      <td>1.00085</td>\n",
       "      <td>1.99906</td>\n",
       "      <td>1.00123</td>\n",
       "      <td>1.01452</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2964</th>\n",
       "      <td>rs28886421</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.98999</td>\n",
       "      <td>1.99000</td>\n",
       "      <td>1.97996</td>\n",
       "      <td>1.97997</td>\n",
       "      <td>1.98999</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.99045</td>\n",
       "      <td>1.98954</td>\n",
       "      <td>1.97913</td>\n",
       "      <td>1.98168</td>\n",
       "      <td>1.98959</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.98954</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2965 rows × 1711 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          SNP_ID REF ALT  10442701  11444465  21112011  20297791  11395417  \\\n",
       "0     rs72777665   T   A   2.00000   2.00000   2.00000   2.00000   2.00000   \n",
       "1     rs35345466   A   T   0.93633   0.26569   0.03411   0.93413   0.95241   \n",
       "2     rs11257183   T   C   1.00000   2.00000   2.00000   1.00000   2.00000   \n",
       "3     rs16914105   C   T   1.00000   2.00000   2.00000   2.00000   2.00000   \n",
       "4      rs7917618   G   A   2.00000   1.99896   1.99931   1.99999   1.99999   \n",
       "...          ...  ..  ..       ...       ...       ...       ...       ...   \n",
       "2960  rs36137267   T   C   1.02021   1.01901   0.98175   1.01908   1.99010   \n",
       "2961   rs6606291   C   T   1.89330   1.88121   1.88121   1.88121   1.88121   \n",
       "2962  rs12337435   C   A   1.00350   1.00536   1.99921   0.02163   1.99959   \n",
       "2963  rs12343184   T   C   1.00089   1.00113   1.99937   0.01343   1.99959   \n",
       "2964  rs28886421   A   G   2.00000   1.98999   1.99000   1.97996   1.97997   \n",
       "\n",
       "      21174303  10669174  ...  50406183  50401426  10263208  50109053  \\\n",
       "0      2.00000   2.00000  ...   1.28960   1.28864   2.00000   1.28953   \n",
       "1      0.96472   0.94774  ...   0.93148   0.00000   0.92957   0.00000   \n",
       "2      2.00000   2.00000  ...   2.00000   2.00000   1.00000   1.00000   \n",
       "3      2.00000   2.00000  ...   2.00000   2.00000   2.00000   2.00000   \n",
       "4      1.99907   2.00000  ...   1.99911   1.99965   2.00000   2.00000   \n",
       "...        ...       ...  ...       ...       ...       ...       ...   \n",
       "2960   0.97431   1.97976  ...   1.94194   0.03687   0.07033   0.09149   \n",
       "2961   1.89330   1.88121  ...   1.88489   1.89351   1.88489   1.89351   \n",
       "2962   0.02076   1.00962  ...   1.01501   1.99907   1.01535   1.99949   \n",
       "2963   0.01337   1.00651  ...   1.01438   1.99923   1.01441   1.99969   \n",
       "2964   1.98999   2.00000  ...   1.99045   1.98954   1.97913   1.98168   \n",
       "\n",
       "      50104396  33006377  50402693  11349821  11099004  29821047  \n",
       "0      2.00000   2.00000   2.00000   2.00000   1.28922   2.00000  \n",
       "1      0.14522   1.84339   0.91914   1.15320   0.16229   0.16842  \n",
       "2      2.00000   2.00000   1.00000   2.00000   2.00000   1.00000  \n",
       "3      2.00000   1.00000   2.00000   1.00000   2.00000   2.00000  \n",
       "4      2.00000   1.99992   1.99649   2.00000   1.99894   1.99820  \n",
       "...        ...       ...       ...       ...       ...       ...  \n",
       "2960   1.05846   1.01997   1.06632   1.85975   1.02561   1.97771  \n",
       "2961   1.88489   1.88489   1.88484   1.89351   1.89351   1.88489  \n",
       "2962   1.01626   1.00376   1.99873   1.00267   1.01543   2.00000  \n",
       "2963   1.01460   1.00085   1.99906   1.00123   1.01452   2.00000  \n",
       "2964   1.98959   2.00000   1.98954   2.00000   2.00000   2.00000  \n",
       "\n",
       "[2965 rows x 1711 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2745eaa2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "24670e17",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Clinical data for mapping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "39e03e88",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>projid</th>\n",
       "      <th>Study</th>\n",
       "      <th>msex</th>\n",
       "      <th>educ</th>\n",
       "      <th>race</th>\n",
       "      <th>spanish</th>\n",
       "      <th>apoe_genotype</th>\n",
       "      <th>age_at_visit_max</th>\n",
       "      <th>age_first_ad_dx</th>\n",
       "      <th>age_death</th>\n",
       "      <th>cts_mmse30_first_ad_dx</th>\n",
       "      <th>cts_mmse30_lv</th>\n",
       "      <th>pmi</th>\n",
       "      <th>braaksc</th>\n",
       "      <th>ceradsc</th>\n",
       "      <th>cogdx</th>\n",
       "      <th>dcfdx_lv</th>\n",
       "      <th>individualID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10101589</td>\n",
       "      <td>ROS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>18.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>9.916667</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>R6939144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>86767530</td>\n",
       "      <td>MAP</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>18.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>6.500000</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>R3893503</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9650662</td>\n",
       "      <td>MAP</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.850000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>R8937093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>50402855</td>\n",
       "      <td>MAP</td>\n",
       "      <td>0.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>27.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>R7139444</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20544321</td>\n",
       "      <td>ROS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>90+</td>\n",
       "      <td>90+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>13.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>R4971237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3579</th>\n",
       "      <td>22207815</td>\n",
       "      <td>ROS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>57.653661875427787</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>R5306025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3580</th>\n",
       "      <td>22207941</td>\n",
       "      <td>ROS</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>56.651608487337441</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>27.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>R6142763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3581</th>\n",
       "      <td>49333806</td>\n",
       "      <td>MAP</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>56.599589322381931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>30.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>R4468842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3582</th>\n",
       "      <td>59720188</td>\n",
       "      <td>MAP</td>\n",
       "      <td>0.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.622861054072551</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>R9446033</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3583</th>\n",
       "      <td>90214403</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R5615105</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3584 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        projid Study  msex  educ  race  spanish  apoe_genotype  \\\n",
       "0     10101589   ROS   1.0  20.0   1.0      2.0           34.0   \n",
       "1     86767530   MAP   0.0  10.0   1.0      2.0           33.0   \n",
       "2      9650662   MAP   0.0  15.0   1.0      2.0           23.0   \n",
       "3     50402855   MAP   0.0  21.0   1.0      2.0           33.0   \n",
       "4     20544321   ROS   0.0  16.0   1.0      2.0           23.0   \n",
       "...        ...   ...   ...   ...   ...      ...            ...   \n",
       "3579  22207815   ROS   0.0  18.0   2.0      2.0           23.0   \n",
       "3580  22207941   ROS   0.0  16.0   2.0      2.0           34.0   \n",
       "3581  49333806   MAP   0.0  12.0   2.0      2.0            NaN   \n",
       "3582  59720188   MAP   0.0  13.0   1.0      1.0            NaN   \n",
       "3583  90214403   NaN   NaN   NaN   NaN      NaN            NaN   \n",
       "\n",
       "        age_at_visit_max age_first_ad_dx age_death  cts_mmse30_first_ad_dx  \\\n",
       "0                    90+             90+       90+                    18.0   \n",
       "1                    90+             90+       90+                    18.0   \n",
       "2                    90+             90+       90+                     0.0   \n",
       "3                    90+             NaN       NaN                     NaN   \n",
       "4                    90+             90+       NaN                    13.0   \n",
       "...                  ...             ...       ...                     ...   \n",
       "3579  57.653661875427787             NaN       NaN                     NaN   \n",
       "3580  56.651608487337441             NaN       NaN                     NaN   \n",
       "3581  56.599589322381931             NaN       NaN                     NaN   \n",
       "3582  54.622861054072551             NaN       NaN                     NaN   \n",
       "3583                 NaN             NaN       NaN                     NaN   \n",
       "\n",
       "      cts_mmse30_lv       pmi  braaksc  ceradsc  cogdx  dcfdx_lv individualID  \n",
       "0               5.0  9.916667      4.0      2.0    4.0       4.0     R6939144  \n",
       "1              10.0  6.500000      4.0      2.0    4.0       4.0     R3893503  \n",
       "2               0.0  3.850000      3.0      2.0    4.0       4.0     R8937093  \n",
       "3              27.0       NaN      NaN      NaN    NaN       1.0     R7139444  \n",
       "4              14.0       NaN      NaN      NaN    NaN       4.0     R4971237  \n",
       "...             ...       ...      ...      ...    ...       ...          ...  \n",
       "3579           29.0       NaN      NaN      NaN    NaN       1.0     R5306025  \n",
       "3580           27.0       NaN      NaN      NaN    NaN       1.0     R6142763  \n",
       "3581           30.0       NaN      NaN      NaN    NaN       1.0     R4468842  \n",
       "3582           29.0       NaN      NaN      NaN    NaN       1.0     R9446033  \n",
       "3583            NaN       NaN      NaN      NaN    NaN       NaN     R5615105  \n",
       "\n",
       "[3584 rows x 18 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ros_clinical = pd.read_csv(\"/home/vmottaqi/rnaseq_synapse/ad_shap_gwas/dosage_imputed_vcf_files/ROSMAP_clinical.csv\")\n",
    "ros_clinical"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3057c300",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create a mapping dictionary from projid to individualID\n",
    "mapping_dict = pd.Series(ros_clinical['individualID'].values, index=ros_clinical['projid']).to_dict()\n",
    "\n",
    "# Ensure all keys and values in the mapping dictionary are strings\n",
    "mapping_dict = {str(key): str(value) for key, value in mapping_dict.items()}\n",
    "\n",
    "# Ensure all column names in df1 are strings\n",
    "df.columns = df.columns.astype(str)\n",
    "\n",
    "# Rename the columns of the first DataFrame using the mapping dictionary\n",
    "df.rename(columns=mapping_dict, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ebd63564",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remove the REF and ALT columns\n",
    "df.drop(columns=['REF', 'ALT'], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "26401251",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP_ID</th>\n",
       "      <th>R3627269</th>\n",
       "      <th>R6655598</th>\n",
       "      <th>R7540069</th>\n",
       "      <th>R5789564</th>\n",
       "      <th>R7047807</th>\n",
       "      <th>R8375980</th>\n",
       "      <th>R4641987</th>\n",
       "      <th>R1801421</th>\n",
       "      <th>R3322663</th>\n",
       "      <th>...</th>\n",
       "      <th>R6835201</th>\n",
       "      <th>R5313591</th>\n",
       "      <th>R5647192</th>\n",
       "      <th>R7914294</th>\n",
       "      <th>R3450006</th>\n",
       "      <th>R3649358</th>\n",
       "      <th>R8075334</th>\n",
       "      <th>R4847810</th>\n",
       "      <th>R4131479</th>\n",
       "      <th>R8093392</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs72777665</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.28960</td>\n",
       "      <td>1.28864</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.28953</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.28922</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs35345466</td>\n",
       "      <td>0.93633</td>\n",
       "      <td>0.26569</td>\n",
       "      <td>0.03411</td>\n",
       "      <td>0.93413</td>\n",
       "      <td>0.95241</td>\n",
       "      <td>0.96472</td>\n",
       "      <td>0.94774</td>\n",
       "      <td>0.00008</td>\n",
       "      <td>0.08625</td>\n",
       "      <td>...</td>\n",
       "      <td>0.93148</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.92957</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.14522</td>\n",
       "      <td>1.84339</td>\n",
       "      <td>0.91914</td>\n",
       "      <td>1.15320</td>\n",
       "      <td>0.16229</td>\n",
       "      <td>0.16842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs11257183</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs16914105</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs7917618</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.99896</td>\n",
       "      <td>1.99931</td>\n",
       "      <td>1.99999</td>\n",
       "      <td>1.99999</td>\n",
       "      <td>1.99907</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.99859</td>\n",
       "      <td>1.99950</td>\n",
       "      <td>...</td>\n",
       "      <td>1.99911</td>\n",
       "      <td>1.99965</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.99992</td>\n",
       "      <td>1.99649</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.99894</td>\n",
       "      <td>1.99820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2960</th>\n",
       "      <td>rs36137267</td>\n",
       "      <td>1.02021</td>\n",
       "      <td>1.01901</td>\n",
       "      <td>0.98175</td>\n",
       "      <td>1.01908</td>\n",
       "      <td>1.99010</td>\n",
       "      <td>0.97431</td>\n",
       "      <td>1.97976</td>\n",
       "      <td>1.03732</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.94194</td>\n",
       "      <td>0.03687</td>\n",
       "      <td>0.07033</td>\n",
       "      <td>0.09149</td>\n",
       "      <td>1.05846</td>\n",
       "      <td>1.01997</td>\n",
       "      <td>1.06632</td>\n",
       "      <td>1.85975</td>\n",
       "      <td>1.02561</td>\n",
       "      <td>1.97771</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2961</th>\n",
       "      <td>rs6606291</td>\n",
       "      <td>1.89330</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>1.89330</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>1.89318</td>\n",
       "      <td>...</td>\n",
       "      <td>1.88489</td>\n",
       "      <td>1.89351</td>\n",
       "      <td>1.88489</td>\n",
       "      <td>1.89351</td>\n",
       "      <td>1.88489</td>\n",
       "      <td>1.88489</td>\n",
       "      <td>1.88484</td>\n",
       "      <td>1.89351</td>\n",
       "      <td>1.89351</td>\n",
       "      <td>1.88489</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2962</th>\n",
       "      <td>rs12337435</td>\n",
       "      <td>1.00350</td>\n",
       "      <td>1.00536</td>\n",
       "      <td>1.99921</td>\n",
       "      <td>0.02163</td>\n",
       "      <td>1.99959</td>\n",
       "      <td>0.02076</td>\n",
       "      <td>1.00962</td>\n",
       "      <td>1.99981</td>\n",
       "      <td>1.01454</td>\n",
       "      <td>...</td>\n",
       "      <td>1.01501</td>\n",
       "      <td>1.99907</td>\n",
       "      <td>1.01535</td>\n",
       "      <td>1.99949</td>\n",
       "      <td>1.01626</td>\n",
       "      <td>1.00376</td>\n",
       "      <td>1.99873</td>\n",
       "      <td>1.00267</td>\n",
       "      <td>1.01543</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2963</th>\n",
       "      <td>rs12343184</td>\n",
       "      <td>1.00089</td>\n",
       "      <td>1.00113</td>\n",
       "      <td>1.99937</td>\n",
       "      <td>0.01343</td>\n",
       "      <td>1.99959</td>\n",
       "      <td>0.01337</td>\n",
       "      <td>1.00651</td>\n",
       "      <td>1.99981</td>\n",
       "      <td>1.00401</td>\n",
       "      <td>...</td>\n",
       "      <td>1.01438</td>\n",
       "      <td>1.99923</td>\n",
       "      <td>1.01441</td>\n",
       "      <td>1.99969</td>\n",
       "      <td>1.01460</td>\n",
       "      <td>1.00085</td>\n",
       "      <td>1.99906</td>\n",
       "      <td>1.00123</td>\n",
       "      <td>1.01452</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2964</th>\n",
       "      <td>rs28886421</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.98999</td>\n",
       "      <td>1.99000</td>\n",
       "      <td>1.97996</td>\n",
       "      <td>1.97997</td>\n",
       "      <td>1.98999</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.99087</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.99045</td>\n",
       "      <td>1.98954</td>\n",
       "      <td>1.97913</td>\n",
       "      <td>1.98168</td>\n",
       "      <td>1.98959</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.98954</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2965 rows × 1709 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          SNP_ID  R3627269  R6655598  R7540069  R5789564  R7047807  R8375980  \\\n",
       "0     rs72777665   2.00000   2.00000   2.00000   2.00000   2.00000   2.00000   \n",
       "1     rs35345466   0.93633   0.26569   0.03411   0.93413   0.95241   0.96472   \n",
       "2     rs11257183   1.00000   2.00000   2.00000   1.00000   2.00000   2.00000   \n",
       "3     rs16914105   1.00000   2.00000   2.00000   2.00000   2.00000   2.00000   \n",
       "4      rs7917618   2.00000   1.99896   1.99931   1.99999   1.99999   1.99907   \n",
       "...          ...       ...       ...       ...       ...       ...       ...   \n",
       "2960  rs36137267   1.02021   1.01901   0.98175   1.01908   1.99010   0.97431   \n",
       "2961   rs6606291   1.89330   1.88121   1.88121   1.88121   1.88121   1.89330   \n",
       "2962  rs12337435   1.00350   1.00536   1.99921   0.02163   1.99959   0.02076   \n",
       "2963  rs12343184   1.00089   1.00113   1.99937   0.01343   1.99959   0.01337   \n",
       "2964  rs28886421   2.00000   1.98999   1.99000   1.97996   1.97997   1.98999   \n",
       "\n",
       "      R4641987  R1801421  R3322663  ...  R6835201  R5313591  R5647192  \\\n",
       "0      2.00000   2.00000   2.00000  ...   1.28960   1.28864   2.00000   \n",
       "1      0.94774   0.00008   0.08625  ...   0.93148   0.00000   0.92957   \n",
       "2      2.00000   1.00000   1.00000  ...   2.00000   2.00000   1.00000   \n",
       "3      2.00000   2.00000   2.00000  ...   2.00000   2.00000   2.00000   \n",
       "4      2.00000   1.99859   1.99950  ...   1.99911   1.99965   2.00000   \n",
       "...        ...       ...       ...  ...       ...       ...       ...   \n",
       "2960   1.97976   1.03732   2.00000  ...   1.94194   0.03687   0.07033   \n",
       "2961   1.88121   1.88121   1.89318  ...   1.88489   1.89351   1.88489   \n",
       "2962   1.00962   1.99981   1.01454  ...   1.01501   1.99907   1.01535   \n",
       "2963   1.00651   1.99981   1.00401  ...   1.01438   1.99923   1.01441   \n",
       "2964   2.00000   1.99087   2.00000  ...   1.99045   1.98954   1.97913   \n",
       "\n",
       "      R7914294  R3450006  R3649358  R8075334  R4847810  R4131479  R8093392  \n",
       "0      1.28953   2.00000   2.00000   2.00000   2.00000   1.28922   2.00000  \n",
       "1      0.00000   0.14522   1.84339   0.91914   1.15320   0.16229   0.16842  \n",
       "2      1.00000   2.00000   2.00000   1.00000   2.00000   2.00000   1.00000  \n",
       "3      2.00000   2.00000   1.00000   2.00000   1.00000   2.00000   2.00000  \n",
       "4      2.00000   2.00000   1.99992   1.99649   2.00000   1.99894   1.99820  \n",
       "...        ...       ...       ...       ...       ...       ...       ...  \n",
       "2960   0.09149   1.05846   1.01997   1.06632   1.85975   1.02561   1.97771  \n",
       "2961   1.89351   1.88489   1.88489   1.88484   1.89351   1.89351   1.88489  \n",
       "2962   1.99949   1.01626   1.00376   1.99873   1.00267   1.01543   2.00000  \n",
       "2963   1.99969   1.01460   1.00085   1.99906   1.00123   1.01452   2.00000  \n",
       "2964   1.98168   1.98959   2.00000   1.98954   2.00000   2.00000   2.00000  \n",
       "\n",
       "[2965 rows x 1709 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fc67e74c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "62ff9ac4",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# Transpose the dataframe\n",
    "transposed_df = df.set_index('SNP_ID').transpose()\n",
    "\n",
    "# Save the transposed dataframe to a pickle file\n",
    "pickle_file_path = 'rosmap_gwas_snps_matrix.pkl'\n",
    "transposed_df.to_pickle(pickle_file_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "86536224",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>SNP_ID</th>\n",
       "      <th>rs72777665</th>\n",
       "      <th>rs35345466</th>\n",
       "      <th>rs11257183</th>\n",
       "      <th>rs16914105</th>\n",
       "      <th>rs7917618</th>\n",
       "      <th>rs2271564</th>\n",
       "      <th>rs7099368</th>\n",
       "      <th>rs920964</th>\n",
       "      <th>rs2458663</th>\n",
       "      <th>rs1177701</th>\n",
       "      <th>...</th>\n",
       "      <th>rs34674752</th>\n",
       "      <th>rs12550729</th>\n",
       "      <th>rs12549149</th>\n",
       "      <th>rs11136254</th>\n",
       "      <th>rs34173062</th>\n",
       "      <th>rs36137267</th>\n",
       "      <th>rs6606291</th>\n",
       "      <th>rs12337435</th>\n",
       "      <th>rs12343184</th>\n",
       "      <th>rs28886421</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>R3627269</th>\n",
       "      <td>2.00000</td>\n",
       "      <td>0.93633</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.67278</td>\n",
       "      <td>1.99141</td>\n",
       "      <td>1.99307</td>\n",
       "      <td>1.99163</td>\n",
       "      <td>1.88831</td>\n",
       "      <td>1.02021</td>\n",
       "      <td>1.89330</td>\n",
       "      <td>1.00350</td>\n",
       "      <td>1.00089</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R6655598</th>\n",
       "      <td>2.00000</td>\n",
       "      <td>0.26569</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.99896</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.99729</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.99990</td>\n",
       "      <td>...</td>\n",
       "      <td>1.95673</td>\n",
       "      <td>1.98918</td>\n",
       "      <td>1.98849</td>\n",
       "      <td>1.98779</td>\n",
       "      <td>1.90800</td>\n",
       "      <td>1.01901</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>1.00536</td>\n",
       "      <td>1.00113</td>\n",
       "      <td>1.98999</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R7540069</th>\n",
       "      <td>2.00000</td>\n",
       "      <td>0.03411</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.99931</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.99272</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.99984</td>\n",
       "      <td>...</td>\n",
       "      <td>1.95586</td>\n",
       "      <td>1.99446</td>\n",
       "      <td>1.99502</td>\n",
       "      <td>1.99358</td>\n",
       "      <td>1.83608</td>\n",
       "      <td>0.98175</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>1.99921</td>\n",
       "      <td>1.99937</td>\n",
       "      <td>1.99000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R5789564</th>\n",
       "      <td>2.00000</td>\n",
       "      <td>0.93413</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.99999</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.21695</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.00008</td>\n",
       "      <td>...</td>\n",
       "      <td>1.96835</td>\n",
       "      <td>1.00098</td>\n",
       "      <td>0.99897</td>\n",
       "      <td>0.99872</td>\n",
       "      <td>1.92303</td>\n",
       "      <td>1.01908</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>0.02163</td>\n",
       "      <td>0.01343</td>\n",
       "      <td>1.97996</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R7047807</th>\n",
       "      <td>2.00000</td>\n",
       "      <td>0.95241</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.99999</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.21641</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.99986</td>\n",
       "      <td>...</td>\n",
       "      <td>1.97187</td>\n",
       "      <td>1.99207</td>\n",
       "      <td>1.99313</td>\n",
       "      <td>1.99163</td>\n",
       "      <td>1.83777</td>\n",
       "      <td>1.99010</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>1.99959</td>\n",
       "      <td>1.99959</td>\n",
       "      <td>1.97997</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R3649358</th>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.84339</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.99992</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.99999</td>\n",
       "      <td>...</td>\n",
       "      <td>1.96900</td>\n",
       "      <td>1.07577</td>\n",
       "      <td>1.09246</td>\n",
       "      <td>1.09309</td>\n",
       "      <td>1.93213</td>\n",
       "      <td>1.01997</td>\n",
       "      <td>1.88489</td>\n",
       "      <td>1.00376</td>\n",
       "      <td>1.00085</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R8075334</th>\n",
       "      <td>2.00000</td>\n",
       "      <td>0.91914</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.99649</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00012</td>\n",
       "      <td>...</td>\n",
       "      <td>1.89073</td>\n",
       "      <td>1.99686</td>\n",
       "      <td>1.99477</td>\n",
       "      <td>1.99593</td>\n",
       "      <td>1.86481</td>\n",
       "      <td>1.06632</td>\n",
       "      <td>1.88484</td>\n",
       "      <td>1.99873</td>\n",
       "      <td>1.99906</td>\n",
       "      <td>1.98954</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R4847810</th>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.15320</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.96669</td>\n",
       "      <td>1.99747</td>\n",
       "      <td>1.99496</td>\n",
       "      <td>1.99598</td>\n",
       "      <td>1.84143</td>\n",
       "      <td>1.85975</td>\n",
       "      <td>1.89351</td>\n",
       "      <td>1.00267</td>\n",
       "      <td>1.00123</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R4131479</th>\n",
       "      <td>1.28922</td>\n",
       "      <td>0.16229</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.99894</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.99910</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.99850</td>\n",
       "      <td>...</td>\n",
       "      <td>1.82715</td>\n",
       "      <td>1.99638</td>\n",
       "      <td>1.99503</td>\n",
       "      <td>1.99590</td>\n",
       "      <td>1.88012</td>\n",
       "      <td>1.02561</td>\n",
       "      <td>1.89351</td>\n",
       "      <td>1.01543</td>\n",
       "      <td>1.01452</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R8093392</th>\n",
       "      <td>2.00000</td>\n",
       "      <td>0.16842</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.99820</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00012</td>\n",
       "      <td>...</td>\n",
       "      <td>1.96594</td>\n",
       "      <td>1.99732</td>\n",
       "      <td>1.99521</td>\n",
       "      <td>1.99551</td>\n",
       "      <td>1.83273</td>\n",
       "      <td>1.97771</td>\n",
       "      <td>1.88489</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1708 rows × 2965 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "SNP_ID    rs72777665  rs35345466  rs11257183  rs16914105  rs7917618  \\\n",
       "R3627269     2.00000     0.93633         1.0         1.0    2.00000   \n",
       "R6655598     2.00000     0.26569         2.0         2.0    1.99896   \n",
       "R7540069     2.00000     0.03411         2.0         2.0    1.99931   \n",
       "R5789564     2.00000     0.93413         1.0         2.0    1.99999   \n",
       "R7047807     2.00000     0.95241         2.0         2.0    1.99999   \n",
       "...              ...         ...         ...         ...        ...   \n",
       "R3649358     2.00000     1.84339         2.0         1.0    1.99992   \n",
       "R8075334     2.00000     0.91914         1.0         2.0    1.99649   \n",
       "R4847810     2.00000     1.15320         2.0         1.0    2.00000   \n",
       "R4131479     1.28922     0.16229         2.0         2.0    1.99894   \n",
       "R8093392     2.00000     0.16842         1.0         2.0    1.99820   \n",
       "\n",
       "SNP_ID    rs2271564  rs7099368  rs920964  rs2458663  rs1177701  ...  \\\n",
       "R3627269        1.0        0.0   2.00000        1.0    1.00000  ...   \n",
       "R6655598        1.0        1.0   1.99729        1.0    0.99990  ...   \n",
       "R7540069        2.0        2.0   1.99272        2.0    1.99984  ...   \n",
       "R5789564        2.0        1.0   1.21695        1.0    1.00008  ...   \n",
       "R7047807        1.0        1.0   1.21641        1.0    0.99986  ...   \n",
       "...             ...        ...       ...        ...        ...  ...   \n",
       "R3649358        2.0        1.0   2.00000        1.0    0.99999  ...   \n",
       "R8075334        1.0        1.0   2.00000        0.0    0.00012  ...   \n",
       "R4847810        2.0        0.0   2.00000        1.0    0.00000  ...   \n",
       "R4131479        1.0        1.0   1.99910        2.0    1.99850  ...   \n",
       "R8093392        1.0        1.0   2.00000        1.0    0.00012  ...   \n",
       "\n",
       "SNP_ID    rs34674752  rs12550729  rs12549149  rs11136254  rs34173062  \\\n",
       "R3627269     1.67278     1.99141     1.99307     1.99163     1.88831   \n",
       "R6655598     1.95673     1.98918     1.98849     1.98779     1.90800   \n",
       "R7540069     1.95586     1.99446     1.99502     1.99358     1.83608   \n",
       "R5789564     1.96835     1.00098     0.99897     0.99872     1.92303   \n",
       "R7047807     1.97187     1.99207     1.99313     1.99163     1.83777   \n",
       "...              ...         ...         ...         ...         ...   \n",
       "R3649358     1.96900     1.07577     1.09246     1.09309     1.93213   \n",
       "R8075334     1.89073     1.99686     1.99477     1.99593     1.86481   \n",
       "R4847810     1.96669     1.99747     1.99496     1.99598     1.84143   \n",
       "R4131479     1.82715     1.99638     1.99503     1.99590     1.88012   \n",
       "R8093392     1.96594     1.99732     1.99521     1.99551     1.83273   \n",
       "\n",
       "SNP_ID    rs36137267  rs6606291  rs12337435  rs12343184  rs28886421  \n",
       "R3627269     1.02021    1.89330     1.00350     1.00089     2.00000  \n",
       "R6655598     1.01901    1.88121     1.00536     1.00113     1.98999  \n",
       "R7540069     0.98175    1.88121     1.99921     1.99937     1.99000  \n",
       "R5789564     1.01908    1.88121     0.02163     0.01343     1.97996  \n",
       "R7047807     1.99010    1.88121     1.99959     1.99959     1.97997  \n",
       "...              ...        ...         ...         ...         ...  \n",
       "R3649358     1.01997    1.88489     1.00376     1.00085     2.00000  \n",
       "R8075334     1.06632    1.88484     1.99873     1.99906     1.98954  \n",
       "R4847810     1.85975    1.89351     1.00267     1.00123     2.00000  \n",
       "R4131479     1.02561    1.89351     1.01543     1.01452     2.00000  \n",
       "R8093392     1.97771    1.88489     2.00000     2.00000     2.00000  \n",
       "\n",
       "[1708 rows x 2965 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df = pd.read_pickle(\"rosmap_gwas_snps_matrix.pkl\")\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c9d13007",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "6184785b",
   "metadata": {},
   "source": [
    "## 3. Plink files and obtaining iid-snps matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "66a54ba5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BIM and FAM files loaded successfully.\n",
      "BIM DataFrame processed successfully.\n",
      "Genotype array created successfully.\n",
      "Number of SNPs to keep: 428\n",
      "Total number of SNPs: 750173\n",
      "Proportion of SNPs to keep: 0.00\n",
      "Final DataFrame shape: (1708, 428)\n",
      "DataFrame has been saved in chunks as csv files.\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from pyplink import PyPlink\n",
    "import pickle\n",
    "\n",
    "\n",
    "plink_file_prefix = \"/home/vmottaqi/rnaseq_synapse/ad_shap_gwas/Rosmap_plink_files/ROSMAP_arrayGenotype\"\n",
    "output_dir = \"/home/vmottaqi/rnaseq_synapse/ad_shap_gwas/Rosmap_plink_files/\"\n",
    "\n",
    "\n",
    "with open('/home/vmottaqi/rnaseq_synapse/ad_shap_gwas/snps_list/gene_variants_dict_AD_filt_phenotype.pkl', 'rb') as f:\n",
    "    gene_variants_dict = pickle.load(f)\n",
    "    \n",
    "# Create a set to store unique SNP identifiers\n",
    "valid_snp_ids_phenotype = set()\n",
    "# Loop through each list in the dictionary and add all items to the set\n",
    "for snp_list in gene_variants_dict.values():\n",
    "    valid_snp_ids_phenotype.update(snp_list)\n",
    "\n",
    " \n",
    "        \n",
    "try:\n",
    "    # Load BIM and FAM files\n",
    "    with PyPlink(plink_file_prefix) as plink:\n",
    "        bim = plink.get_bim()\n",
    "        fam = plink.get_fam()\n",
    "\n",
    "    # Create DataFrames but only keep relevant columns\n",
    "    bim_df = pd.DataFrame(bim)\n",
    "    fam_df = pd.DataFrame(fam) \n",
    "\n",
    "    \n",
    "    print(\"BIM and FAM files loaded successfully.\")\n",
    "\n",
    "    # Handle Missing rsIDs and '*' Alleles\n",
    "    bim_df['a1'] = bim_df['a1'].replace('*', 'unknown')\n",
    "    bim_df['a2'] = bim_df['a2'].replace('*', 'unknown')\n",
    "\n",
    "    # Assuming the custom format is [Chromosome Number]:[Coordinate], as stated in README\n",
    "    # bim_df['snp'] = bim_df.apply(lambda row: f\"chr{row['chrom']}:{row['pos']}\" if pd.isna(row['snp']) else row['snp'], axis=1)\n",
    "\n",
    "    print(\"BIM DataFrame processed successfully.\")\n",
    "\n",
    "    # Extract Genotypes\n",
    "    genotypes = []\n",
    "    with PyPlink(plink_file_prefix) as plink:\n",
    "        for gt in plink:\n",
    "            genotypes.append(gt[1])  # Store genotype array\n",
    "\n",
    "    genotypes_arr = np.array(genotypes, dtype='int8').T  # Transpose to have individuals as rows and SNPs as columns\n",
    "    print(\"Genotype array created successfully.\")\n",
    "\n",
    "    snps_without_missing_values_mask = np.ones(len(genotypes_arr[0]), dtype=bool)\n",
    "\n",
    "    # Identify SNPs with known alleles (i.e., not 'unknown')\n",
    "    known_alleles_mask = (bim_df['a1'] != 'unknown') & (bim_df['a2'] != 'unknown')\n",
    "\n",
    "    # Combine the masks to identify SNPs to keep\n",
    "    snps_to_keep_mask = snps_without_missing_values_mask & known_alleles_mask & bim_df.index.isin(valid_snp_ids_phenotype) # It's a boolean!\n",
    "\n",
    "    # Check the number of SNPs to keep (number of True values)\n",
    "    num_snps_to_keep = np.sum(snps_to_keep_mask)\n",
    "    print(f\"Number of SNPs to keep: {num_snps_to_keep}\")\n",
    "\n",
    "    # Check the total number of SNPs\n",
    "    total_snps = len(snps_to_keep_mask)\n",
    "    print(f\"Total number of SNPs: {total_snps}\")\n",
    "\n",
    "    # Optionally, print the proportion of SNPs to keep\n",
    "    proportion_snps_to_keep = num_snps_to_keep / total_snps\n",
    "    print(f\"Proportion of SNPs to keep: {proportion_snps_to_keep:.2f}\")\n",
    "\n",
    "    # Get the list of SNP IDs to keep\n",
    "    snps_to_keep = bim_df.index[snps_to_keep_mask].tolist()\n",
    "\n",
    "    # Filter the genotype array to keep only SNPs without any -1 values and with known alleles\n",
    "    genotypes_filtered = genotypes_arr[:, snps_to_keep_mask]\n",
    "\n",
    "    # Create a DataFrame with EPR identifiers as rows and SNPs as columns\n",
    "    final_df = pd.DataFrame(genotypes_filtered, index=fam_df['iid'], columns=snps_to_keep)\n",
    "\n",
    "\n",
    "    print(f\"Final DataFrame shape: {final_df.shape}\")\n",
    "\n",
    "    # Extract base name for the output files\n",
    "    base_name = \"ROSMAP_genotypes\"\n",
    "\n",
    "    final_df.to_csv(f'{output_dir}{base_name}.csv')\n",
    "    \n",
    "    print(\"DataFrame has been saved in chunks as csv files.\")\n",
    "\n",
    "except Exception as e:\n",
    "    print(f\"An error occurred: {e}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "945d68f7",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "165906ba",
   "metadata": {},
   "source": [
    "## 4. Filtering iid-snps matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f216856b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filtering based on the filtered samples (samples of interests from clustering)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "105c9946",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>SNP_ID</th>\n",
       "      <th>rs72777665</th>\n",
       "      <th>rs35345466</th>\n",
       "      <th>rs11257183</th>\n",
       "      <th>rs16914105</th>\n",
       "      <th>rs7917618</th>\n",
       "      <th>rs2271564</th>\n",
       "      <th>rs7099368</th>\n",
       "      <th>rs920964</th>\n",
       "      <th>rs2458663</th>\n",
       "      <th>rs1177701</th>\n",
       "      <th>...</th>\n",
       "      <th>rs34674752</th>\n",
       "      <th>rs12550729</th>\n",
       "      <th>rs12549149</th>\n",
       "      <th>rs11136254</th>\n",
       "      <th>rs34173062</th>\n",
       "      <th>rs36137267</th>\n",
       "      <th>rs6606291</th>\n",
       "      <th>rs12337435</th>\n",
       "      <th>rs12343184</th>\n",
       "      <th>rs28886421</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>R3627269</th>\n",
       "      <td>2.00000</td>\n",
       "      <td>0.93633</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.67278</td>\n",
       "      <td>1.99141</td>\n",
       "      <td>1.99307</td>\n",
       "      <td>1.99163</td>\n",
       "      <td>1.88831</td>\n",
       "      <td>1.02021</td>\n",
       "      <td>1.89330</td>\n",
       "      <td>1.00350</td>\n",
       "      <td>1.00089</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R6655598</th>\n",
       "      <td>2.00000</td>\n",
       "      <td>0.26569</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.99896</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.99729</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.99990</td>\n",
       "      <td>...</td>\n",
       "      <td>1.95673</td>\n",
       "      <td>1.98918</td>\n",
       "      <td>1.98849</td>\n",
       "      <td>1.98779</td>\n",
       "      <td>1.90800</td>\n",
       "      <td>1.01901</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>1.00536</td>\n",
       "      <td>1.00113</td>\n",
       "      <td>1.98999</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R7540069</th>\n",
       "      <td>2.00000</td>\n",
       "      <td>0.03411</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.99931</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.99272</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.99984</td>\n",
       "      <td>...</td>\n",
       "      <td>1.95586</td>\n",
       "      <td>1.99446</td>\n",
       "      <td>1.99502</td>\n",
       "      <td>1.99358</td>\n",
       "      <td>1.83608</td>\n",
       "      <td>0.98175</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>1.99921</td>\n",
       "      <td>1.99937</td>\n",
       "      <td>1.99000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R5789564</th>\n",
       "      <td>2.00000</td>\n",
       "      <td>0.93413</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.99999</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.21695</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.00008</td>\n",
       "      <td>...</td>\n",
       "      <td>1.96835</td>\n",
       "      <td>1.00098</td>\n",
       "      <td>0.99897</td>\n",
       "      <td>0.99872</td>\n",
       "      <td>1.92303</td>\n",
       "      <td>1.01908</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>0.02163</td>\n",
       "      <td>0.01343</td>\n",
       "      <td>1.97996</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R7047807</th>\n",
       "      <td>2.00000</td>\n",
       "      <td>0.95241</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.99999</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.21641</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.99986</td>\n",
       "      <td>...</td>\n",
       "      <td>1.97187</td>\n",
       "      <td>1.99207</td>\n",
       "      <td>1.99313</td>\n",
       "      <td>1.99163</td>\n",
       "      <td>1.83777</td>\n",
       "      <td>1.99010</td>\n",
       "      <td>1.88121</td>\n",
       "      <td>1.99959</td>\n",
       "      <td>1.99959</td>\n",
       "      <td>1.97997</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R3649358</th>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.84339</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.99992</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.99999</td>\n",
       "      <td>...</td>\n",
       "      <td>1.96900</td>\n",
       "      <td>1.07577</td>\n",
       "      <td>1.09246</td>\n",
       "      <td>1.09309</td>\n",
       "      <td>1.93213</td>\n",
       "      <td>1.01997</td>\n",
       "      <td>1.88489</td>\n",
       "      <td>1.00376</td>\n",
       "      <td>1.00085</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R8075334</th>\n",
       "      <td>2.00000</td>\n",
       "      <td>0.91914</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.99649</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00012</td>\n",
       "      <td>...</td>\n",
       "      <td>1.89073</td>\n",
       "      <td>1.99686</td>\n",
       "      <td>1.99477</td>\n",
       "      <td>1.99593</td>\n",
       "      <td>1.86481</td>\n",
       "      <td>1.06632</td>\n",
       "      <td>1.88484</td>\n",
       "      <td>1.99873</td>\n",
       "      <td>1.99906</td>\n",
       "      <td>1.98954</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R4847810</th>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.15320</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.96669</td>\n",
       "      <td>1.99747</td>\n",
       "      <td>1.99496</td>\n",
       "      <td>1.99598</td>\n",
       "      <td>1.84143</td>\n",
       "      <td>1.85975</td>\n",
       "      <td>1.89351</td>\n",
       "      <td>1.00267</td>\n",
       "      <td>1.00123</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R4131479</th>\n",
       "      <td>1.28922</td>\n",
       "      <td>0.16229</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.99894</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.99910</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.99850</td>\n",
       "      <td>...</td>\n",
       "      <td>1.82715</td>\n",
       "      <td>1.99638</td>\n",
       "      <td>1.99503</td>\n",
       "      <td>1.99590</td>\n",
       "      <td>1.88012</td>\n",
       "      <td>1.02561</td>\n",
       "      <td>1.89351</td>\n",
       "      <td>1.01543</td>\n",
       "      <td>1.01452</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R8093392</th>\n",
       "      <td>2.00000</td>\n",
       "      <td>0.16842</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.99820</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00012</td>\n",
       "      <td>...</td>\n",
       "      <td>1.96594</td>\n",
       "      <td>1.99732</td>\n",
       "      <td>1.99521</td>\n",
       "      <td>1.99551</td>\n",
       "      <td>1.83273</td>\n",
       "      <td>1.97771</td>\n",
       "      <td>1.88489</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "      <td>2.00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1708 rows × 2965 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "SNP_ID    rs72777665  rs35345466  rs11257183  rs16914105  rs7917618  \\\n",
       "R3627269     2.00000     0.93633         1.0         1.0    2.00000   \n",
       "R6655598     2.00000     0.26569         2.0         2.0    1.99896   \n",
       "R7540069     2.00000     0.03411         2.0         2.0    1.99931   \n",
       "R5789564     2.00000     0.93413         1.0         2.0    1.99999   \n",
       "R7047807     2.00000     0.95241         2.0         2.0    1.99999   \n",
       "...              ...         ...         ...         ...        ...   \n",
       "R3649358     2.00000     1.84339         2.0         1.0    1.99992   \n",
       "R8075334     2.00000     0.91914         1.0         2.0    1.99649   \n",
       "R4847810     2.00000     1.15320         2.0         1.0    2.00000   \n",
       "R4131479     1.28922     0.16229         2.0         2.0    1.99894   \n",
       "R8093392     2.00000     0.16842         1.0         2.0    1.99820   \n",
       "\n",
       "SNP_ID    rs2271564  rs7099368  rs920964  rs2458663  rs1177701  ...  \\\n",
       "R3627269        1.0        0.0   2.00000        1.0    1.00000  ...   \n",
       "R6655598        1.0        1.0   1.99729        1.0    0.99990  ...   \n",
       "R7540069        2.0        2.0   1.99272        2.0    1.99984  ...   \n",
       "R5789564        2.0        1.0   1.21695        1.0    1.00008  ...   \n",
       "R7047807        1.0        1.0   1.21641        1.0    0.99986  ...   \n",
       "...             ...        ...       ...        ...        ...  ...   \n",
       "R3649358        2.0        1.0   2.00000        1.0    0.99999  ...   \n",
       "R8075334        1.0        1.0   2.00000        0.0    0.00012  ...   \n",
       "R4847810        2.0        0.0   2.00000        1.0    0.00000  ...   \n",
       "R4131479        1.0        1.0   1.99910        2.0    1.99850  ...   \n",
       "R8093392        1.0        1.0   2.00000        1.0    0.00012  ...   \n",
       "\n",
       "SNP_ID    rs34674752  rs12550729  rs12549149  rs11136254  rs34173062  \\\n",
       "R3627269     1.67278     1.99141     1.99307     1.99163     1.88831   \n",
       "R6655598     1.95673     1.98918     1.98849     1.98779     1.90800   \n",
       "R7540069     1.95586     1.99446     1.99502     1.99358     1.83608   \n",
       "R5789564     1.96835     1.00098     0.99897     0.99872     1.92303   \n",
       "R7047807     1.97187     1.99207     1.99313     1.99163     1.83777   \n",
       "...              ...         ...         ...         ...         ...   \n",
       "R3649358     1.96900     1.07577     1.09246     1.09309     1.93213   \n",
       "R8075334     1.89073     1.99686     1.99477     1.99593     1.86481   \n",
       "R4847810     1.96669     1.99747     1.99496     1.99598     1.84143   \n",
       "R4131479     1.82715     1.99638     1.99503     1.99590     1.88012   \n",
       "R8093392     1.96594     1.99732     1.99521     1.99551     1.83273   \n",
       "\n",
       "SNP_ID    rs36137267  rs6606291  rs12337435  rs12343184  rs28886421  \n",
       "R3627269     1.02021    1.89330     1.00350     1.00089     2.00000  \n",
       "R6655598     1.01901    1.88121     1.00536     1.00113     1.98999  \n",
       "R7540069     0.98175    1.88121     1.99921     1.99937     1.99000  \n",
       "R5789564     1.01908    1.88121     0.02163     0.01343     1.97996  \n",
       "R7047807     1.99010    1.88121     1.99959     1.99959     1.97997  \n",
       "...              ...        ...         ...         ...         ...  \n",
       "R3649358     1.01997    1.88489     1.00376     1.00085     2.00000  \n",
       "R8075334     1.06632    1.88484     1.99873     1.99906     1.98954  \n",
       "R4847810     1.85975    1.89351     1.00267     1.00123     2.00000  \n",
       "R4131479     1.02561    1.89351     1.01543     1.01452     2.00000  \n",
       "R8093392     1.97771    1.88489     2.00000     2.00000     2.00000  \n",
       "\n",
       "[1708 rows x 2965 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "matrix = pd.read_pickle(\"rosmap_gwas_snps_matrix.pkl\")\n",
    "matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3bbb836e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fad66979",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>individualID</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>tissue</th>\n",
       "      <th>clusters</th>\n",
       "      <th>race</th>\n",
       "      <th>spanish</th>\n",
       "      <th>apoe4_allele</th>\n",
       "      <th>sex</th>\n",
       "      <th>batch</th>\n",
       "      <th>pmi</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSG00000287815</th>\n",
       "      <th>ENSG00000287900</th>\n",
       "      <th>ENSG00000287925</th>\n",
       "      <th>ENSG00000287978</th>\n",
       "      <th>ENSG00000287985</th>\n",
       "      <th>ENSG00000288033</th>\n",
       "      <th>ENSG00000288048</th>\n",
       "      <th>ENSG00000288049</th>\n",
       "      <th>ENSG00000288062</th>\n",
       "      <th>ENSG00000288107</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>366_120502</th>\n",
       "      <td>R9809661</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>3_1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>17.416667</td>\n",
       "      <td>...</td>\n",
       "      <td>0.139377</td>\n",
       "      <td>0.833608</td>\n",
       "      <td>1.092943</td>\n",
       "      <td>1.707821</td>\n",
       "      <td>1.329693</td>\n",
       "      <td>0.960361</td>\n",
       "      <td>0.552581</td>\n",
       "      <td>-5.109539</td>\n",
       "      <td>3.378009</td>\n",
       "      <td>1.674681</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52_120416</th>\n",
       "      <td>R9809441</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>3_2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>4.450000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.635089</td>\n",
       "      <td>0.925383</td>\n",
       "      <td>0.934533</td>\n",
       "      <td>0.243795</td>\n",
       "      <td>-0.272757</td>\n",
       "      <td>0.441563</td>\n",
       "      <td>1.028262</td>\n",
       "      <td>-4.070924</td>\n",
       "      <td>2.869645</td>\n",
       "      <td>1.041287</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>493_120515</th>\n",
       "      <td>R9771008</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>3_0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>3.083333</td>\n",
       "      <td>...</td>\n",
       "      <td>1.428795</td>\n",
       "      <td>1.639106</td>\n",
       "      <td>0.797550</td>\n",
       "      <td>1.391738</td>\n",
       "      <td>1.096291</td>\n",
       "      <td>1.636495</td>\n",
       "      <td>0.642124</td>\n",
       "      <td>-0.998250</td>\n",
       "      <td>3.053631</td>\n",
       "      <td>1.721891</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>654_120529</th>\n",
       "      <td>R9254464</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>3_2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>7.500000</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.432613</td>\n",
       "      <td>1.611975</td>\n",
       "      <td>0.995050</td>\n",
       "      <td>1.056309</td>\n",
       "      <td>1.156011</td>\n",
       "      <td>-0.180843</td>\n",
       "      <td>1.239391</td>\n",
       "      <td>-5.325863</td>\n",
       "      <td>3.342447</td>\n",
       "      <td>2.135636</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76_120417</th>\n",
       "      <td>R9137173</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>3_1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>5.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.326804</td>\n",
       "      <td>1.561044</td>\n",
       "      <td>-0.693230</td>\n",
       "      <td>2.267006</td>\n",
       "      <td>1.651049</td>\n",
       "      <td>0.810119</td>\n",
       "      <td>0.634079</td>\n",
       "      <td>-4.042548</td>\n",
       "      <td>3.783358</td>\n",
       "      <td>1.414760</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RISK_437</th>\n",
       "      <td>R9425423</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>3_1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>17</td>\n",
       "      <td>16.516667</td>\n",
       "      <td>...</td>\n",
       "      <td>0.831489</td>\n",
       "      <td>1.606163</td>\n",
       "      <td>1.261024</td>\n",
       "      <td>0.970229</td>\n",
       "      <td>0.885443</td>\n",
       "      <td>0.769497</td>\n",
       "      <td>-1.164219</td>\n",
       "      <td>-4.505437</td>\n",
       "      <td>3.107846</td>\n",
       "      <td>1.777936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RISK_46_rerun</th>\n",
       "      <td>R6998518</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>6_1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>17</td>\n",
       "      <td>15.666667</td>\n",
       "      <td>...</td>\n",
       "      <td>1.210712</td>\n",
       "      <td>1.852605</td>\n",
       "      <td>0.874080</td>\n",
       "      <td>1.637173</td>\n",
       "      <td>-0.111618</td>\n",
       "      <td>1.523786</td>\n",
       "      <td>0.782975</td>\n",
       "      <td>-5.904522</td>\n",
       "      <td>1.779158</td>\n",
       "      <td>2.831715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RISK_50_rerun</th>\n",
       "      <td>R1438115</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>6_1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>17</td>\n",
       "      <td>6.750000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.865323</td>\n",
       "      <td>1.656371</td>\n",
       "      <td>1.406023</td>\n",
       "      <td>1.376175</td>\n",
       "      <td>0.480853</td>\n",
       "      <td>1.533203</td>\n",
       "      <td>-0.687184</td>\n",
       "      <td>-5.655950</td>\n",
       "      <td>2.795944</td>\n",
       "      <td>1.290207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RISK_51_rerun</th>\n",
       "      <td>R1753622</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>3_2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>17</td>\n",
       "      <td>7.266667</td>\n",
       "      <td>...</td>\n",
       "      <td>0.394446</td>\n",
       "      <td>1.370487</td>\n",
       "      <td>0.705370</td>\n",
       "      <td>0.248185</td>\n",
       "      <td>-0.039540</td>\n",
       "      <td>0.556972</td>\n",
       "      <td>0.763065</td>\n",
       "      <td>-6.118090</td>\n",
       "      <td>2.972036</td>\n",
       "      <td>1.743947</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RISK_67_rerun</th>\n",
       "      <td>R6925335</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>3_3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>38.333333</td>\n",
       "      <td>...</td>\n",
       "      <td>1.053959</td>\n",
       "      <td>1.893846</td>\n",
       "      <td>1.672508</td>\n",
       "      <td>2.134204</td>\n",
       "      <td>0.823070</td>\n",
       "      <td>1.460374</td>\n",
       "      <td>-0.035521</td>\n",
       "      <td>1.167405</td>\n",
       "      <td>3.214965</td>\n",
       "      <td>1.984574</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>300 rows × 4018 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              individualID  diagnosis  tissue clusters  race  spanish  \\\n",
       "366_120502        R9809661          1       3      3_1   1.0      2.0   \n",
       "52_120416         R9809441          1       3      3_2   1.0      2.0   \n",
       "493_120515        R9771008          1       3      3_0   1.0      2.0   \n",
       "654_120529        R9254464          1       3      3_2   1.0      2.0   \n",
       "76_120417         R9137173          1       3      3_1   1.0      2.0   \n",
       "...                    ...        ...     ...      ...   ...      ...   \n",
       "RISK_437          R9425423          1       3      3_1   1.0      2.0   \n",
       "RISK_46_rerun     R6998518          1       6      6_1   1.0      2.0   \n",
       "RISK_50_rerun     R1438115          1       6      6_1   1.0      2.0   \n",
       "RISK_51_rerun     R1753622          1       3      3_2   1.0      2.0   \n",
       "RISK_67_rerun     R6925335          1       3      3_3   1.0      2.0   \n",
       "\n",
       "               apoe4_allele  sex  batch        pmi  ...  ENSG00000287815  \\\n",
       "366_120502                1    1      7  17.416667  ...         0.139377   \n",
       "52_120416                 0    1      6   4.450000  ...         0.635089   \n",
       "493_120515                0    0      4   3.083333  ...         1.428795   \n",
       "654_120529                0    1      5   7.500000  ...        -0.432613   \n",
       "76_120417                 1    1      4   5.000000  ...         0.326804   \n",
       "...                     ...  ...    ...        ...  ...              ...   \n",
       "RISK_437                  0    1     17  16.516667  ...         0.831489   \n",
       "RISK_46_rerun             1    1     17  15.666667  ...         1.210712   \n",
       "RISK_50_rerun             0    1     17   6.750000  ...         0.865323   \n",
       "RISK_51_rerun             2    1     17   7.266667  ...         0.394446   \n",
       "RISK_67_rerun             0    0     17  38.333333  ...         1.053959   \n",
       "\n",
       "               ENSG00000287900  ENSG00000287925  ENSG00000287978  \\\n",
       "366_120502            0.833608         1.092943         1.707821   \n",
       "52_120416             0.925383         0.934533         0.243795   \n",
       "493_120515            1.639106         0.797550         1.391738   \n",
       "654_120529            1.611975         0.995050         1.056309   \n",
       "76_120417             1.561044        -0.693230         2.267006   \n",
       "...                        ...              ...              ...   \n",
       "RISK_437              1.606163         1.261024         0.970229   \n",
       "RISK_46_rerun         1.852605         0.874080         1.637173   \n",
       "RISK_50_rerun         1.656371         1.406023         1.376175   \n",
       "RISK_51_rerun         1.370487         0.705370         0.248185   \n",
       "RISK_67_rerun         1.893846         1.672508         2.134204   \n",
       "\n",
       "               ENSG00000287985  ENSG00000288033  ENSG00000288048  \\\n",
       "366_120502            1.329693         0.960361         0.552581   \n",
       "52_120416            -0.272757         0.441563         1.028262   \n",
       "493_120515            1.096291         1.636495         0.642124   \n",
       "654_120529            1.156011        -0.180843         1.239391   \n",
       "76_120417             1.651049         0.810119         0.634079   \n",
       "...                        ...              ...              ...   \n",
       "RISK_437              0.885443         0.769497        -1.164219   \n",
       "RISK_46_rerun        -0.111618         1.523786         0.782975   \n",
       "RISK_50_rerun         0.480853         1.533203        -0.687184   \n",
       "RISK_51_rerun        -0.039540         0.556972         0.763065   \n",
       "RISK_67_rerun         0.823070         1.460374        -0.035521   \n",
       "\n",
       "               ENSG00000288049 ENSG00000288062  ENSG00000288107  \n",
       "366_120502           -5.109539        3.378009         1.674681  \n",
       "52_120416            -4.070924        2.869645         1.041287  \n",
       "493_120515           -0.998250        3.053631         1.721891  \n",
       "654_120529           -5.325863        3.342447         2.135636  \n",
       "76_120417            -4.042548        3.783358         1.414760  \n",
       "...                        ...             ...              ...  \n",
       "RISK_437             -4.505437        3.107846         1.777936  \n",
       "RISK_46_rerun        -5.904522        1.779158         2.831715  \n",
       "RISK_50_rerun        -5.655950        2.795944         1.290207  \n",
       "RISK_51_rerun        -6.118090        2.972036         1.743947  \n",
       "RISK_67_rerun         1.167405        3.214965         1.984574  \n",
       "\n",
       "[300 rows x 4018 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "rosmap = pd.read_csv(\"../ROSMAP_shap_samples_v12_4k_with_clusters.csv\", index_col=0)\n",
    "rosmap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "06a27097",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of unique values in column 'XXX': 174\n"
     ]
    }
   ],
   "source": [
    "unique_values_count = rosmap['individualID'].nunique()\n",
    "print(f\"Number of unique values in column 'XXX': {unique_values_count}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9830360a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of common values (individuals between rosmap and gwas matrix): 127\n"
     ]
    }
   ],
   "source": [
    "# Intersection of Specimen ID\n",
    "\n",
    "first_col_matrix = matrix.index.astype(str).tolist()\n",
    "first_col_rosmap = rosmap.iloc[:, 0].astype(str).tolist()\n",
    "intersection = set(first_col_matrix).intersection(first_col_rosmap)\n",
    "\n",
    "# Print the number of common values\n",
    "print(f'Number of common values (individuals between rosmap and gwas matrix): {len(intersection)}')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "739c26a1",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e52edd85",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter the SNPs dataset to contain only individuals present in rosmap data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "017a9579",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SNP_ID    rs72777665  rs35345466  rs11257183  rs16914105  rs7917618  \\\n",
      "R8882846     2.00000     1.87074         2.0         2.0    1.99940   \n",
      "R9823938     2.00000     0.93168         2.0         2.0    1.99988   \n",
      "R9064073     1.32430     0.00001         2.0         2.0    1.99811   \n",
      "R8059669     2.00000     0.94354         2.0         2.0    2.00000   \n",
      "R5955028     1.28995     0.00000         1.0         2.0    1.99906   \n",
      "...              ...         ...         ...         ...        ...   \n",
      "R6411801     2.00000     0.39161         2.0         2.0    1.99970   \n",
      "R1154454     2.00000     0.91812         1.0         2.0    1.99860   \n",
      "R4388056     2.00000     0.00006         2.0         2.0    1.99817   \n",
      "R7289081     1.27963     0.91665         2.0         2.0    2.00000   \n",
      "R8647890     2.00000     1.84956         2.0         2.0    1.99892   \n",
      "\n",
      "SNP_ID    rs2271564  rs7099368  rs920964  rs2458663  rs1177701  ...  \\\n",
      "R8882846        2.0        2.0   1.99299        1.0    0.99995  ...   \n",
      "R9823938        1.0        1.0   1.99636        1.0    0.99996  ...   \n",
      "R9064073        1.0        1.0   1.99875        1.0    0.99985  ...   \n",
      "R8059669        2.0        1.0   1.19032        1.0    0.99924  ...   \n",
      "R5955028        1.0        1.0   1.99829        1.0    0.99930  ...   \n",
      "...             ...        ...       ...        ...        ...  ...   \n",
      "R6411801        1.0        1.0   1.99850        1.0    0.99953  ...   \n",
      "R1154454        2.0        2.0   1.99618        2.0    1.99832  ...   \n",
      "R4388056        1.0        1.0   1.99748        1.0    0.99984  ...   \n",
      "R7289081        2.0        0.0   2.00000        0.0    0.00029  ...   \n",
      "R8647890        1.0        1.0   1.99711        1.0    0.99987  ...   \n",
      "\n",
      "SNP_ID    rs34674752  rs12550729  rs12549149  rs11136254  rs34173062  \\\n",
      "R8882846     1.97098     1.99245     1.99342     1.99191     1.83652   \n",
      "R9823938     1.87909     1.02815     1.04426     1.04467     1.92940   \n",
      "R9064073     1.85465     1.99635     1.99558     1.99554     1.85657   \n",
      "R8059669     1.88461     1.99685     1.99575     1.99595     1.85235   \n",
      "R5955028     1.89261     1.99625     1.99509     1.99512     1.85162   \n",
      "...              ...         ...         ...         ...         ...   \n",
      "R6411801     1.96954     1.03483     1.03643     1.03641     1.91152   \n",
      "R1154454     1.89254     1.99626     1.99268     1.99390     1.87930   \n",
      "R4388056     1.85661     1.99514     1.99405     1.99415     1.86772   \n",
      "R7289081     1.92364     1.88556     1.88636     1.88593     1.89398   \n",
      "R8647890     1.75589     1.99338     1.99538     1.99399     1.85971   \n",
      "\n",
      "SNP_ID    rs36137267  rs6606291  rs12337435  rs12343184  rs28886421  \n",
      "R8882846     0.19670    1.89330     1.00483     1.00133     2.00000  \n",
      "R9823938     1.99527    1.90010     1.99993     1.99999     1.98052  \n",
      "R9064073     1.00758    1.88619     1.99917     1.99921     1.99292  \n",
      "R8059669     0.10968    1.89351     0.97233     0.97255     1.98956  \n",
      "R5955028     0.09067    1.88484     1.99981     1.99981     2.00000  \n",
      "...              ...        ...         ...         ...         ...  \n",
      "R6411801     0.05936    1.89518     1.99998     1.99997     1.98740  \n",
      "R1154454     1.99030    1.88489     1.05079     1.04924     2.00000  \n",
      "R4388056     0.07189    1.88619     1.01965     1.00239     1.99281  \n",
      "R7289081     0.99844    1.88121     0.00869     0.00232     1.98999  \n",
      "R8647890     1.93522    1.89330     1.99929     1.99944     1.99009  \n",
      "\n",
      "[127 rows x 2965 columns]\n",
      "Filtered SNP matrix saved as 'rosmap_gwas_snps_matrix_filtered.pkl'. Total individuals retained: 127\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Load your data\n",
    "clusters_df = pd.read_csv('../ROSMAP_shap_samples_v12_4k_with_clusters.csv', index_col=0, dtype={'individualID': str})\n",
    "snps_df = pd.read_pickle('rosmap_gwas_snps_matrix.pkl')\n",
    "snps_df.index = snps_df.index.map(str)  # Convert index to string to match clusters_df\n",
    "\n",
    "# Find the intersection of individual IDs between clusters and SNPs data\n",
    "mutual_individuals = set(clusters_df['individualID']).intersection(snps_df.index)\n",
    "\n",
    "# Filter the SNP matrix to include only mutual individual IDs\n",
    "filtered_snps_df = snps_df.loc[list(mutual_individuals)]\n",
    "print(filtered_snps_df)\n",
    "\n",
    "# Save the filtered DataFrame to a new pickle file\n",
    "filtered_snps_file_path = 'rosmap_gwas_snps_matrix_filtered.pkl'\n",
    "filtered_snps_df.to_pickle(filtered_snps_file_path)\n",
    "\n",
    "print(f\"Filtered SNP matrix saved as '{filtered_snps_file_path}'. Total individuals retained: {len(mutual_individuals)}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fe38a5bb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter the dataset to remove columns with one unique value"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3a6a2360",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of columns with only one unique value: 397\n",
      "Filtered DataFrame saved as 'rosmap_gwas_snps_matrix_filtered_final.pkl'.\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "df = pd.read_pickle(\"rosmap_gwas_snps_matrix_filtered.pkl\")\n",
    "\n",
    "# Find columns with only one unique value\n",
    "single_value_columns = [col for col in df.columns if df[col].nunique() == 1]\n",
    "print(f\"Number of columns with only one unique value: {len(single_value_columns)}\")\n",
    "\n",
    "# Remove these columns from the DataFrame\n",
    "df_filtered = df.drop(columns=single_value_columns)\n",
    "\n",
    "# Save the filtered DataFrame to a pickle file\n",
    "df_filtered.to_pickle('rosmap_gwas_snps_matrix_filtered_final.pkl')\n",
    "\n",
    "print(\"Filtered DataFrame saved as 'rosmap_gwas_snps_matrix_filtered_final.pkl'.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b088a928",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "536d7d83",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f8e2ab68",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3a8c6144",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "268ca3ee",
   "metadata": {},
   "source": [
    "# GWAS data (MSBB)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0bd45ee3",
   "metadata": {},
   "source": [
    "## 1. Plink files and obtaining iid-snps matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8f5ae84f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# plink files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3ce3b1fb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BIM and FAM files loaded successfully.\n",
      "BIM DataFrame processed successfully.\n",
      "Genotype array created successfully.\n",
      "Number of SNPs to keep: 991\n",
      "Total number of SNPs: 117132\n",
      "Proportion of SNPs to keep: 0.01\n",
      "Final DataFrame shape: (344, 991)\n",
      "DataFrame has been saved in chunks as csv files.\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from pyplink import PyPlink\n",
    "import pickle\n",
    "\n",
    "\n",
    "plink_file_prefix = \"/home/vmottaqi/rnaseq_synapse/ad_shap_gwas/MSBB_plink_files/AMP-AD_MSBB_MSSM_IlluminaHiSeq2500_WES_QCed\"\n",
    "output_dir = \"/home/vmottaqi/rnaseq_synapse/ad_shap_gwas/MSBB_plink_files/\"\n",
    "\n",
    "\n",
    "with open('/home/vmottaqi/rnaseq_synapse/ad_shap_gwas/snps_list/gene_variants_dict_AD.pkl', 'rb') as f:\n",
    "    gene_variants_dict = pickle.load(f)\n",
    "    \n",
    "# Create a set to store unique SNP identifiers\n",
    "valid_snp_ids_phenotype = set()\n",
    "# Loop through each list in the dictionary and add all items to the set\n",
    "for snp_list in gene_variants_dict.values():\n",
    "    valid_snp_ids_phenotype.update(snp_list)\n",
    "\n",
    " \n",
    "        \n",
    "try:\n",
    "    # Load BIM and FAM files\n",
    "    with PyPlink(plink_file_prefix) as plink:\n",
    "        bim = plink.get_bim()\n",
    "        fam = plink.get_fam()\n",
    "\n",
    "    # Create DataFrames but only keep relevant columns\n",
    "    bim_df = pd.DataFrame(bim)\n",
    "    fam_df = pd.DataFrame(fam) \n",
    "\n",
    "    \n",
    "    print(\"BIM and FAM files loaded successfully.\")\n",
    "\n",
    "    # Handle Missing rsIDs and '*' Alleles\n",
    "    bim_df['a1'] = bim_df['a1'].replace('*', 'unknown')\n",
    "    bim_df['a2'] = bim_df['a2'].replace('*', 'unknown')\n",
    "\n",
    "    # Assuming the custom format is [Chromosome Number]:[Coordinate], as stated in README\n",
    "    # bim_df['snp'] = bim_df.apply(lambda row: f\"chr{row['chrom']}:{row['pos']}\" if pd.isna(row['snp']) else row['snp'], axis=1)\n",
    "\n",
    "    print(\"BIM DataFrame processed successfully.\")\n",
    "\n",
    "    # Extract Genotypes\n",
    "    genotypes = []\n",
    "    with PyPlink(plink_file_prefix) as plink:\n",
    "        for gt in plink:\n",
    "            genotypes.append(gt[1])  # Store genotype array\n",
    "\n",
    "    genotypes_arr = np.array(genotypes, dtype='int8').T  # Transpose to have individuals as rows and SNPs as columns\n",
    "    print(\"Genotype array created successfully.\")\n",
    "\n",
    "    # Identify SNPs to keep (i.e., without any -1 values across all samples)\n",
    "    snps_without_missing_values_mask = np.all(genotypes_arr != -1, axis=0)\n",
    "\n",
    "    # Identify SNPs with known alleles (i.e., not 'unknown')\n",
    "    known_alleles_mask = (bim_df['a1'] != 'unknown') & (bim_df['a2'] != 'unknown')\n",
    "\n",
    "    # Combine the masks to identify SNPs to keep\n",
    "    snps_to_keep_mask = snps_without_missing_values_mask & known_alleles_mask & bim_df.index.isin(valid_snp_ids_phenotype) # It's a boolean!\n",
    "\n",
    "    # Check the number of SNPs to keep (number of True values)\n",
    "    num_snps_to_keep = np.sum(snps_to_keep_mask)\n",
    "    print(f\"Number of SNPs to keep: {num_snps_to_keep}\")\n",
    "\n",
    "    # Check the total number of SNPs\n",
    "    total_snps = len(snps_to_keep_mask)\n",
    "    print(f\"Total number of SNPs: {total_snps}\")\n",
    "\n",
    "    # Optionally, print the proportion of SNPs to keep\n",
    "    proportion_snps_to_keep = num_snps_to_keep / total_snps\n",
    "    print(f\"Proportion of SNPs to keep: {proportion_snps_to_keep:.2f}\")\n",
    "\n",
    "    # Get the list of SNP IDs to keep\n",
    "    snps_to_keep = bim_df.index[snps_to_keep_mask].tolist()\n",
    "\n",
    "    # Filter the genotype array to keep only SNPs without any -1 values and with known alleles\n",
    "    genotypes_filtered = genotypes_arr[:, snps_to_keep_mask]\n",
    "\n",
    "    # Create a DataFrame with EPR identifiers as rows and SNPs as columns\n",
    "    final_df = pd.DataFrame(genotypes_filtered, index=fam_df['iid'], columns=snps_to_keep)\n",
    "\n",
    "\n",
    "    print(f\"Final DataFrame shape: {final_df.shape}\")\n",
    "\n",
    "    # Extract base name for the output files\n",
    "    base_name = \"MSBB_genotypes\"\n",
    "\n",
    "    final_df.to_csv(f'{output_dir}{base_name}.csv')\n",
    "    \n",
    "    print(\"DataFrame has been saved in chunks as csv files.\")\n",
    "\n",
    "except Exception as e:\n",
    "    print(f\"An error occurred: {e}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "de299e5c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "9ae4606f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>iid</th>\n",
       "      <th>rs1889854</th>\n",
       "      <th>rs41278972</th>\n",
       "      <th>rs11121172</th>\n",
       "      <th>rs1058766</th>\n",
       "      <th>rs2784735</th>\n",
       "      <th>rs201922249</th>\n",
       "      <th>rs13596</th>\n",
       "      <th>rs74568545</th>\n",
       "      <th>rs2252865</th>\n",
       "      <th>...</th>\n",
       "      <th>rs417301</th>\n",
       "      <th>rs422803</th>\n",
       "      <th>rs400852</th>\n",
       "      <th>rs370850</th>\n",
       "      <th>rs144355331</th>\n",
       "      <th>rs9981768</th>\n",
       "      <th>rs428785</th>\n",
       "      <th>rs409630</th>\n",
       "      <th>rs445784</th>\n",
       "      <th>rs434857</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BM_10_1084</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BM_10_1121</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BM_10_1122</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BM_10_967</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BM_22_1000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>339</th>\n",
       "      <td>hB_DNA_13208</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340</th>\n",
       "      <td>hB_DNA_13223</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>341</th>\n",
       "      <td>hB_DNA_13233</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>342</th>\n",
       "      <td>hB_DNA_13239</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>343</th>\n",
       "      <td>hB_DNA_9064</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>344 rows × 992 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              iid  rs1889854  rs41278972  rs11121172  rs1058766  rs2784735  \\\n",
       "0      BM_10_1084          0           0           0          0          1   \n",
       "1      BM_10_1121          0           0           1          1          1   \n",
       "2      BM_10_1122          0           0           1          1          1   \n",
       "3       BM_10_967          0           0           1          1          1   \n",
       "4      BM_22_1000          0           0           1          1          1   \n",
       "..            ...        ...         ...         ...        ...        ...   \n",
       "339  hB_DNA_13208          1           0           0          0          2   \n",
       "340  hB_DNA_13223          0           0           1          1          1   \n",
       "341  hB_DNA_13233          0           0           1          1          1   \n",
       "342  hB_DNA_13239          0           0           1          1          1   \n",
       "343   hB_DNA_9064          1           0           0          0          1   \n",
       "\n",
       "     rs201922249  rs13596  rs74568545  rs2252865  ...  rs417301  rs422803  \\\n",
       "0              0        0           0          0  ...         0         0   \n",
       "1              0        1           0          1  ...         0         0   \n",
       "2              0        1           0          1  ...         1         1   \n",
       "3              0        1           0          1  ...         0         0   \n",
       "4              0        1           0          1  ...         1         1   \n",
       "..           ...      ...         ...        ...  ...       ...       ...   \n",
       "339            0        0           0          0  ...         1         1   \n",
       "340            1        1           0          1  ...         0         0   \n",
       "341            0        1           0          1  ...         1         1   \n",
       "342            0        1           0          1  ...         0         0   \n",
       "343            0        0           0          0  ...         0         0   \n",
       "\n",
       "     rs400852  rs370850  rs144355331  rs9981768  rs428785  rs409630  rs445784  \\\n",
       "0           1         0            0          0         0         0         0   \n",
       "1           1         0            0          0         0         1         0   \n",
       "2           1         1            0          0         1         1         1   \n",
       "3           2         0            0          0         0         1         0   \n",
       "4           1         1            0          0         1         1         1   \n",
       "..        ...       ...          ...        ...       ...       ...       ...   \n",
       "339         0         1            0          0         1         0         1   \n",
       "340         0         0            0          1         0         0         0   \n",
       "341         0         1            0          0         1         0         1   \n",
       "342         0         0            0          0         0         0         0   \n",
       "343         0         0            0          0         0         0         0   \n",
       "\n",
       "     rs434857  \n",
       "0           0  \n",
       "1           0  \n",
       "2           1  \n",
       "3           0  \n",
       "4           1  \n",
       "..        ...  \n",
       "339         1  \n",
       "340         0  \n",
       "341         1  \n",
       "342         0  \n",
       "343         0  \n",
       "\n",
       "[344 rows x 992 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv(\"/home/vmottaqi/rnaseq_synapse/ad_shap_gwas/MSBB_plink_files/MSBB_genotypes.csv\")\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8d083a37",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0af5b924",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "afef9315",
   "metadata": {},
   "source": [
    "# GWAS data (Mayo)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5eec93c6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "b1bec9c0",
   "metadata": {},
   "source": [
    "## 1. Plink files and obtaining iid-snps matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7eec1761",
   "metadata": {},
   "outputs": [],
   "source": [
    "# If the code fails, don't transpose it at first (the genotype array)!\n",
    "# I ran it on Mercury and it worked just fine.\n",
    "# Needs revision. Go to the next code"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "87b5d0a7",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from pyplink import PyPlink\n",
    "import pickle\n",
    "\n",
    "\n",
    "plink_file_prefix = \"/home/vmottaqi/rnaseq_synapse/ad_shap_gwas/Mayo_plink_files/MayoRNAseq_RNAseq_Genome-Wide_Genotypes_HRCimputed\"\n",
    "output_dir = \"/home/vmottaqi/rnaseq_synapse/ad_shap_gwas/Mayo_plink_files/\"\n",
    "\n",
    "\n",
    "with open('/home/vmottaqi/rnaseq_synapse/ad_shap_gwas/snps_list/gene_variants_dict_AD_filt_phenotype.pkl', 'rb') as f:\n",
    "    gene_variants_dict = pickle.load(f)\n",
    "    \n",
    "# Create a set to store unique SNP identifiers\n",
    "valid_snp_ids_phenotype = set()\n",
    "# Loop through each list in the dictionary and add all items to the set\n",
    "for snp_list in gene_variants_dict.values():\n",
    "    valid_snp_ids_phenotype.update(snp_list)\n",
    "\n",
    " \n",
    "        \n",
    "try:\n",
    "    # Load BIM and FAM files\n",
    "    with PyPlink(plink_file_prefix) as plink:\n",
    "        bim = plink.get_bim()\n",
    "        fam = plink.get_fam()\n",
    "\n",
    "    # Create DataFrames but only keep relevant columns\n",
    "    bim_df = pd.DataFrame(bim)\n",
    "    fam_df = pd.DataFrame(fam) \n",
    "\n",
    "    \n",
    "    print(\"BIM and FAM files loaded successfully.\")\n",
    "\n",
    "    # Handle Missing rsIDs and '*' Alleles\n",
    "    bim_df['a1'] = bim_df['a1'].replace('*', 'unknown')\n",
    "    bim_df['a2'] = bim_df['a2'].replace('*', 'unknown')\n",
    "\n",
    "    # Assuming the custom format is [Chromosome Number]:[Coordinate], as stated in README\n",
    "    # bim_df['snp'] = bim_df.apply(lambda row: f\"chr{row['chrom']}:{row['pos']}\" if pd.isna(row['snp']) else row['snp'], axis=1)\n",
    "\n",
    "    print(\"BIM DataFrame processed successfully.\")\n",
    "\n",
    "    # Extract Genotypes\n",
    "    genotypes = []\n",
    "    with PyPlink(plink_file_prefix) as plink:\n",
    "        for gt in plink:\n",
    "            genotypes.append(gt[1])  # Store genotype array\n",
    "\n",
    "    genotypes_arr = np.array(genotypes, dtype='int8').T  # Transpose to have individuals as rows and SNPs as columns\n",
    "    print(\"Genotype array created successfully.\")\n",
    "\n",
    "    # Identify SNPs to keep (i.e., without any -1 values across all samples)\n",
    "    snps_without_missing_values_mask = np.all(genotypes_arr != -1, axis=0)\n",
    "\n",
    "    # Identify SNPs with known alleles (i.e., not 'unknown')\n",
    "    known_alleles_mask = (bim_df['a1'] != 'unknown') & (bim_df['a2'] != 'unknown')\n",
    "\n",
    "    # Combine the masks to identify SNPs to keep\n",
    "    snps_to_keep_mask = snps_without_missing_values_mask & known_alleles_mask & bim_df.index.isin(valid_snp_ids_phenotype) # It's a boolean!\n",
    "\n",
    "    # Check the number of SNPs to keep (number of True values)\n",
    "    num_snps_to_keep = np.sum(snps_to_keep_mask)\n",
    "    print(f\"Number of SNPs to keep: {num_snps_to_keep}\")\n",
    "\n",
    "    # Check the total number of SNPs\n",
    "    total_snps = len(snps_to_keep_mask)\n",
    "    print(f\"Total number of SNPs: {total_snps}\")\n",
    "\n",
    "    # Optionally, print the proportion of SNPs to keep\n",
    "    proportion_snps_to_keep = num_snps_to_keep / total_snps\n",
    "    print(f\"Proportion of SNPs to keep: {proportion_snps_to_keep:.2f}\")\n",
    "\n",
    "    # Get the list of SNP IDs to keep\n",
    "    snps_to_keep = bim_df.index[snps_to_keep_mask].tolist()\n",
    "\n",
    "    # Filter the genotype array to keep only SNPs without any -1 values and with known alleles\n",
    "    genotypes_filtered = genotypes_arr[:, snps_to_keep_mask]\n",
    "\n",
    "    # Create a DataFrame with EPR identifiers as rows and SNPs as columns\n",
    "    final_df = pd.DataFrame(genotypes_filtered, index=fam_df['iid'], columns=snps_to_keep)\n",
    "\n",
    "\n",
    "    print(f\"Final DataFrame shape: {final_df.shape}\")\n",
    "\n",
    "    # Extract base name for the output files\n",
    "    base_name = \"MAYO_genotypes\"\n",
    "\n",
    "    final_df.to_csv(f'{output_dir}{base_name}.csv')\n",
    "    \n",
    "    print(\"DataFrame has been saved in chunks as csv files.\")\n",
    "\n",
    "except Exception as e:\n",
    "    print(f\"An error occurred: {e}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "346d8025",
   "metadata": {},
   "outputs": [],
   "source": [
    "# The challenge is that the index of bim_df are not snp IDs, they are locations of the snps. So they need to be mapped\n",
    "# There is another file for mapping purposes\n",
    "# Revised code:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2c12f58d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from pyplink import PyPlink\n",
    "import pickle\n",
    "\n",
    "\n",
    "plink_file_prefix = \"/home/vmottaqi/rnaseq_synapse/Mayo_plink_files/MayoRNAseq_RNAseq_Genome-Wide_Genotypes_HRCimputed\"\n",
    "output_dir = \"/home/vmottaqi/rnaseq_synapse/Mayo_plink_files/\"\n",
    "\n",
    "\n",
    "with open('/home/vmottaqi/rnaseq_synapse/Mayo_plink_files/snps_list/gene_variants_dict_AD_filt_phenotype.pkl', 'rb') as f:\n",
    "    gene_variants_dict = pickle.load(f)\n",
    "\n",
    "\n",
    "snp_mapping_file_path = '/home/vmottaqi/rnaseq_synapse/Mayo_plink_files/mayornaseq_rnaseq_genome-wide_genotypes_hrcimputed_variantidkey.txt'  # Adjust the path to where your SNP mapping file is stored\n",
    "mapping_df = pd.read_csv(snp_mapping_file_path, sep='\\t')  # Assuming tab-separated values\n",
    "\n",
    "# Create the mapping dictionary\n",
    "pos_to_snp = {f\"{row['CHR']}:{row['BP']}\": row['SNP'] for index, row in mapping_df.iterrows()}\n",
    " \n",
    "    \n",
    "# Create a set to store unique SNP identifiers\n",
    "valid_snp_ids_phenotype = set()\n",
    "# Loop through each list in the dictionary and add all items to the set\n",
    "for snp_list in gene_variants_dict.values():\n",
    "    valid_snp_ids_phenotype.update(snp_list)\n",
    "\n",
    " \n",
    "        \n",
    "try:\n",
    "    # Load BIM and FAM files\n",
    "    with PyPlink(plink_file_prefix) as plink:\n",
    "        bim = plink.get_bim()\n",
    "        fam = plink.get_fam()\n",
    "\n",
    "    # Create DataFrames but only keep relevant columns\n",
    "    bim_df = pd.DataFrame(bim)\n",
    "    bim_df.index = [pos_to_snp.get(idx, idx) for idx in bim_df.index]\n",
    "    \n",
    "    fam_df = pd.DataFrame(fam) \n",
    "\n",
    "    print(\"BIM and FAM files loaded successfully.\")\n",
    "\n",
    "    # Handle Missing rsIDs and '*' Alleles\n",
    "    bim_df['a1'] = bim_df['a1'].replace('*', 'unknown')\n",
    "    bim_df['a2'] = bim_df['a2'].replace('*', 'unknown')\n",
    "\n",
    "    # Assuming the custom format is [Chromosome Number]:[Coordinate], as stated in README\n",
    "    # bim_df['snp'] = bim_df.apply(lambda row: f\"chr{row['chrom']}:{row['pos']}\" if pd.isna(row['snp']) else row['snp'], axis=1)\n",
    "\n",
    "    print(\"BIM DataFrame processed successfully.\")\n",
    "\n",
    "    # Extract Genotypes\n",
    "    genotypes = []\n",
    "    with PyPlink(plink_file_prefix) as plink:\n",
    "        for gt in plink:\n",
    "            genotypes.append(gt[1])  # Store genotype array\n",
    "\n",
    "    genotypes_arr = np.array(genotypes, dtype='int8').T  # Transpose to have individuals as rows and SNPs as columns\n",
    "    print(\"Genotype array created successfully.\")\n",
    "\n",
    "    # Identify SNPs to keep (i.e., without any -1 values across all samples)\n",
    "    snps_without_missing_values_mask = np.all(genotypes_arr != -1, axis=0)\n",
    "\n",
    "    # Identify SNPs with known alleles (i.e., not 'unknown')\n",
    "    known_alleles_mask = (bim_df['a1'] != 'unknown') & (bim_df['a2'] != 'unknown')\n",
    "\n",
    "    # Combine the masks to identify SNPs to keep\n",
    "    snps_to_keep_mask = snps_without_missing_values_mask & known_alleles_mask & bim_df.index.isin(valid_snp_ids_phenotype) # It's a boolean!\n",
    "\n",
    "    # Check the number of SNPs to keep (number of True values)\n",
    "    num_snps_to_keep = np.sum(snps_to_keep_mask)\n",
    "    print(f\"Number of SNPs to keep: {num_snps_to_keep}\")\n",
    "\n",
    "    # Check the total number of SNPs\n",
    "    total_snps = len(snps_to_keep_mask)\n",
    "    print(f\"Total number of SNPs: {total_snps}\")\n",
    "\n",
    "    # Optionally, print the proportion of SNPs to keep\n",
    "    proportion_snps_to_keep = num_snps_to_keep / total_snps\n",
    "    print(f\"Proportion of SNPs to keep: {proportion_snps_to_keep:.2f}\")\n",
    "\n",
    "    # Get the list of SNP IDs to keep\n",
    "    snps_to_keep = bim_df.index[snps_to_keep_mask].tolist()\n",
    "\n",
    "    # Filter the genotype array to keep only SNPs without any -1 values and with known alleles\n",
    "    genotypes_filtered = genotypes_arr[:, snps_to_keep_mask]\n",
    "\n",
    "    # Create a DataFrame with EPR identifiers as rows and SNPs as columns\n",
    "    final_df = pd.DataFrame(genotypes_filtered, index=fam_df['iid'], columns=snps_to_keep)\n",
    "\n",
    "\n",
    "    print(f\"Final DataFrame shape: {final_df.shape}\")\n",
    "\n",
    "    # Extract base name for the output files\n",
    "    base_name = \"MAYO_genotypes\"\n",
    "\n",
    "    final_df.to_csv(f'{output_dir}{base_name}.csv')\n",
    "    \n",
    "    print(\"DataFrame has been saved in chunks as csv files.\")\n",
    "\n",
    "except Exception as e:\n",
    "    print(f\"An error occurred: {e}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "51a794a4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c876be2f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "3e25ab66",
   "metadata": {},
   "source": [
    "# Harmonizing three GWAS data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "88fdd43a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "6c7022fd",
   "metadata": {},
   "source": [
    "## 1. checking"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "2397a71e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of columns in MAYO file: 5099\n",
      "Number of columns in MSBB file: 992\n",
      "Number of columns in ROSMAP file: 429\n",
      "Common columns across all three files: 38\n",
      "Common columns between MAYO and MSBB files: 446\n",
      "Common columns between MAYO and ROSMAP files: 427\n",
      "Common columns between MSBB and ROSMAP files: 38\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Load the CSV files into DataFrames\n",
    "df1 = pd.read_csv('/home/vmottaqi/rnaseq_synapse/ad_shap_gwas/genotype_csv_files_aug28/MAYO_genotypes.csv')\n",
    "df2 = pd.read_csv('/home/vmottaqi/rnaseq_synapse/ad_shap_gwas/genotype_csv_files_aug28/MSBB_genotypes.csv')\n",
    "df3 = pd.read_csv('/home/vmottaqi/rnaseq_synapse/ad_shap_gwas/genotype_csv_files_aug28/ROSMAP_genotypes.csv')\n",
    "\n",
    "# Get the column names from each DataFrame\n",
    "cols_df1 = set(df1.columns)\n",
    "cols_df2 = set(df2.columns)\n",
    "cols_df3 = set(df3.columns)\n",
    "\n",
    "# Print the number of columns in each DataFrame\n",
    "print(\"Number of columns in MAYO file:\", len(cols_df1))\n",
    "print(\"Number of columns in MSBB file:\", len(cols_df2))\n",
    "print(\"Number of columns in ROSMAP file:\", len(cols_df3))\n",
    "\n",
    "# Calculate intersections of column names\n",
    "intersection_all = cols_df1 & cols_df2 & cols_df3\n",
    "intersection_12 = cols_df1 & cols_df2\n",
    "intersection_13 = cols_df1 & cols_df3\n",
    "intersection_23 = cols_df2 & cols_df3\n",
    "\n",
    "# Print the number of common columns\n",
    "print(\"Common columns across all three files:\", len(intersection_all))\n",
    "print(\"Common columns between MAYO and MSBB files:\", len(intersection_12))\n",
    "print(\"Common columns between MAYO and ROSMAP files:\", len(intersection_13))\n",
    "print(\"Common columns between MSBB and ROSMAP files:\", len(intersection_23))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "ecc5a3f7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of unique genes corresponding to the common SNP IDs: 23\n"
     ]
    }
   ],
   "source": [
    "# Load the gene-SNP mapping dictionary from pickle file\n",
    "with open('/home/vmottaqi/rnaseq_synapse/ad_shap_gwas/snps_list/gene_variants_dict_AD_filt_phenotype.pkl', 'rb') as f:\n",
    "    gene_snp_dict = pickle.load(f)\n",
    "\n",
    "# Find unique genes corresponding to the intersecting SNP IDs\n",
    "unique_genes = set()\n",
    "for gene, snps in gene_snp_dict.items():\n",
    "    if set(snps).intersection(intersection_all):  # If any SNP in common_snps is also in this gene's SNP list\n",
    "        unique_genes.add(gene)\n",
    "\n",
    "# Print the number of unique genes found\n",
    "print(f\"Number of unique genes corresponding to the common SNP IDs: {len(unique_genes)}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "0af1552a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENSG00000080815\n",
      "ENSG00000136717\n",
      "ENSG00000046604\n",
      "ENSG00000086288\n",
      "ENSG00000120899\n",
      "ENSG00000165806\n",
      "ENSG00000107736\n",
      "ENSG00000029725\n",
      "ENSG00000203710\n",
      "ENSG00000159640\n",
      "ENSG00000119684\n",
      "ENSG00000064687\n",
      "ENSG00000152223\n",
      "ENSG00000197943\n",
      "ENSG00000198796\n",
      "ENSG00000102935\n",
      "ENSG00000168918\n",
      "ENSG00000114805\n",
      "ENSG00000168769\n",
      "ENSG00000141252\n",
      "ENSG00000140090\n",
      "ENSG00000137642\n",
      "ENSG00000186153\n"
     ]
    }
   ],
   "source": [
    "for i in unique_genes:\n",
    "    print(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0a41d7e0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "489d065e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
